{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import main as m\n",
    "import subprocess\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "coverage = 5\n",
    "\n",
    "seq_dict = {\n",
    "    \"numcells\": 10,\n",
    "    \"founder_mult\": 1,\n",
    "    \"resolution\": 960,\n",
    "    \"block_length\": 60\n",
    "}\n",
    "\n",
    "read_dict = {\n",
    "    \"readlen\": 100,\n",
    "    \"outerdist\": 250, \n",
    "    \"numreads\": 100000,\n",
    "    \"readmode\": 1\n",
    "}\n",
    "\n",
    "exp_dict = {\n",
    "    \"p\": \"exp22\"\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Experiment log at /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/log.txt\n",
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n"
     ]
    }
   ],
   "source": [
    "# FIXME later, want these to be high level and then we do\n",
    "# one more level for mixtures\n",
    "outdir = f'{m.OUTDIR}/experiments/{exp_dict[\"p\"]}'\n",
    "_seqdir = f'{outdir}/sequences'\n",
    "_readdir = f'{outdir}/reads'\n",
    "\n",
    "subprocess.call([\"mkdir\", \"-p\", _seqdir])\n",
    "subprocess.call([\"mkdir\", \"-p\", _readdir])\n",
    "\n",
    "log = f'{outdir}/log.txt'\n",
    "m._save_settings(outdir, seq_dict, read_dict, exp_dict)\n",
    "\n",
    "fp, aug_fp, tm, root, celldict, focal = m.genstep(seq_dict, _seqdir, log = log,\n",
    "                                                   easy = True, only_insertions=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Mutation(edgeid='cell6.2', allele=1, num_copies=2, relstart=33142080, refstart=37251840, length=133440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "is_insertion = (tm.num_copies != 0)\n",
    "length = tm.length\n",
    "rl = tm.refstart \n",
    "rr = rl + length\n",
    "tm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.1.33131520.33418560 length: 287040\n",
      "[dwgsim_core] 1 sequences, total length: 287040\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 100000\n",
      "[dwgsim_core] Complete!\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[12], line 7\u001b[0m\n\u001b[1;32m      5\u001b[0m readdir \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m_readdir\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/b\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mi\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m      6\u001b[0m subprocess\u001b[38;5;241m.\u001b[39mcall([\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmkdir\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m-p\u001b[39m\u001b[38;5;124m\"\u001b[39m, readdir])\n\u001b[0;32m----> 7\u001b[0m df, sus_df \u001b[38;5;241m=\u001b[39m \u001b[43mm\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mreadstep\u001b[49m\u001b[43m(\u001b[49m\u001b[43mread_dict\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mreaddir\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mb\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mi\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtm\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maug_fp\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlog\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mlog\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      8\u001b[0m \u001b[38;5;66;03m# if is_insertion:\u001b[39;00m\n\u001b[1;32m      9\u001b[0m \u001b[38;5;66;03m#     fdf = sus_df[(sus_df[\"l\"] >= rr) & (sus_df[\"r\"] <= rl)]\u001b[39;00m\n\u001b[1;32m     10\u001b[0m \u001b[38;5;66;03m# else:\u001b[39;00m\n\u001b[1;32m     11\u001b[0m \u001b[38;5;66;03m#     fdf = sus_df[(sus_df[\"l\"] <= rl) & (sus_df[\"r\"] >= rr)]\u001b[39;00m\n\u001b[1;32m     13\u001b[0m peaks\u001b[38;5;241m.\u001b[39mappend(\u001b[38;5;28mlen\u001b[39m(df[((df[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124ml\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m>\u001b[39m\u001b[38;5;241m=\u001b[39m rl) \u001b[38;5;241m&\u001b[39m (df[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m<\u001b[39m\u001b[38;5;241m=\u001b[39m rr)) \u001b[38;5;241m&\u001b[39m (df[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124ml\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m>\u001b[39m df[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124m\"\u001b[39m])]))\n",
      "File \u001b[0;32m~/Documents/Research/SPATIAL/pipeline/main.py:267\u001b[0m, in \u001b[0;36mreadstep\u001b[0;34m(d, readdir, rp, tm, aug_fp, verbose, log)\u001b[0m\n\u001b[1;32m    265\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mopen\u001b[39m(unmapped, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mw\u001b[39m\u001b[38;5;124m'\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m f_out:\n\u001b[1;32m    266\u001b[0m     _filter_reads(read1, f_out\u001b[38;5;241m.\u001b[39mname, unmapped_reads)\n\u001b[0;32m--> 267\u001b[0m     \u001b[43m_filter_reads\u001b[49m\u001b[43m(\u001b[49m\u001b[43mread2\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mf_out\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mname\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43munmapped_reads\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    269\u001b[0m readprocessing2 \u001b[38;5;241m=\u001b[39m [\n\u001b[1;32m    270\u001b[0m     \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mbwa mem \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mREF\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mdual_read1\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mdual_read2\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m > \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mreaddir\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/target_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mtm\u001b[38;5;241m.\u001b[39medgeid\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.sam\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    271\u001b[0m     \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00msamtools\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m view -bT \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mREF\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mreaddir\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/target_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mtm\u001b[38;5;241m.\u001b[39medgeid\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.sam > \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mreaddir\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/target_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mtm\u001b[38;5;241m.\u001b[39medgeid\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.bam\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    274\u001b[0m     \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{\u001b[39;00msamtools\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m view -F 8 -f 4 \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mreaddir\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/target_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mtm\u001b[38;5;241m.\u001b[39medgeid\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.bam |\u001b[39m\u001b[38;5;124m'\u001b[39m  \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m awk \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m{\u001b[39m\u001b[38;5;124mprint $1, $2, $4, $8, $5}\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m > rmapped.txt\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    275\u001b[0m     \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{\u001b[39;00msamtools\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m view -f 0x2 \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mreaddir\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/target_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mtm\u001b[38;5;241m.\u001b[39medgeid\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.bam |\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m awk \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m{\u001b[39m\u001b[38;5;124mprint $1, $2, $4, $8, $5}\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m > fullmapped.txt\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m    278\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m line \u001b[38;5;129;01min\u001b[39;00m readprocessing2:\n",
      "File \u001b[0;32m~/Documents/Research/SPATIAL/pipeline/main.py:148\u001b[0m, in \u001b[0;36m_filter_reads\u001b[0;34m(input_file, output_file, names_to_keep)\u001b[0m\n\u001b[1;32m    146\u001b[0m seq \u001b[38;5;241m=\u001b[39m f_in\u001b[38;5;241m.\u001b[39mreadline()\n\u001b[1;32m    147\u001b[0m plus \u001b[38;5;241m=\u001b[39m f_in\u001b[38;5;241m.\u001b[39mreadline()\n\u001b[0;32m--> 148\u001b[0m qual \u001b[38;5;241m=\u001b[39m f_in\u001b[38;5;241m.\u001b[39mreadline()\n\u001b[1;32m    149\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m header:\n\u001b[1;32m    150\u001b[0m     \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[0;32m~/miniconda3/envs/cnasim/lib/python3.11/gzip.py:314\u001b[0m, in \u001b[0;36mGzipFile.read1\u001b[0;34m(self, size)\u001b[0m\n\u001b[1;32m    312\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m size \u001b[38;5;241m<\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m    313\u001b[0m     size \u001b[38;5;241m=\u001b[39m io\u001b[38;5;241m.\u001b[39mDEFAULT_BUFFER_SIZE\n\u001b[0;32m--> 314\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_buffer\u001b[38;5;241m.\u001b[39mread1(size)\n",
      "File \u001b[0;32m~/miniconda3/envs/cnasim/lib/python3.11/_compression.py:68\u001b[0m, in \u001b[0;36mDecompressReader.readinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m     66\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mreadinto\u001b[39m(\u001b[38;5;28mself\u001b[39m, b):\n\u001b[1;32m     67\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mmemoryview\u001b[39m(b) \u001b[38;5;28;01mas\u001b[39;00m view, view\u001b[38;5;241m.\u001b[39mcast(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mB\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m byte_view:\n\u001b[0;32m---> 68\u001b[0m         data \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mbyte_view\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     69\u001b[0m         byte_view[:\u001b[38;5;28mlen\u001b[39m(data)] \u001b[38;5;241m=\u001b[39m data\n\u001b[1;32m     70\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(data)\n",
      "File \u001b[0;32m~/miniconda3/envs/cnasim/lib/python3.11/gzip.py:507\u001b[0m, in \u001b[0;36m_GzipReader.read\u001b[0;34m(self, size)\u001b[0m\n\u001b[1;32m    504\u001b[0m \u001b[38;5;66;03m# Read a chunk of data from the file\u001b[39;00m\n\u001b[1;32m    505\u001b[0m buf \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_fp\u001b[38;5;241m.\u001b[39mread(io\u001b[38;5;241m.\u001b[39mDEFAULT_BUFFER_SIZE)\n\u001b[0;32m--> 507\u001b[0m uncompress \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_decompressor\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdecompress\u001b[49m\u001b[43m(\u001b[49m\u001b[43mbuf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msize\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    508\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_decompressor\u001b[38;5;241m.\u001b[39munconsumed_tail \u001b[38;5;241m!=\u001b[39m \u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[1;32m    509\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_fp\u001b[38;5;241m.\u001b[39mprepend(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_decompressor\u001b[38;5;241m.\u001b[39munconsumed_tail)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# single run example\n",
    "signal = []\n",
    "peaks = []\n",
    "for i in range(1):\n",
    "    readdir = f\"{_readdir}/b{i}\"\n",
    "    subprocess.call([\"mkdir\", \"-p\", readdir])\n",
    "    df, sus_df = m.readstep(read_dict, readdir, f\"b{i}\", tm, aug_fp, log = log)\n",
    "    # if is_insertion:\n",
    "    #     fdf = sus_df[(sus_df[\"l\"] >= rr) & (sus_df[\"r\"] <= rl)]\n",
    "    # else:\n",
    "    #     fdf = sus_df[(sus_df[\"l\"] <= rl) & (sus_df[\"r\"] >= rr)]\n",
    "    \n",
    "    peaks.append(len(df[((df[\"l\"] >= rl) & (df[\"r\"] <= rr)) & (df[\"l\"] > df[\"r\"])]))\n",
    "    signal.append(len(df[(df[\"l\"] > rr) & (df[\"r\"] > rl)]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# single run example\n",
    "# signal = []\n",
    "# peaks = []\n",
    "# for i in range(10):\n",
    "#     readdir = f\"{_readdir}/b{i}\"\n",
    "#     subprocess.call([\"mkdir\", \"-p\", readdir])\n",
    "#     df, sus_df = m.readstep(read_dict, readdir, f\"b{i}\", tm, aug_fp, log = log)\n",
    "#     # if is_insertion:\n",
    "#     #     fdf = sus_df[(sus_df[\"l\"] >= rr) & (sus_df[\"r\"] <= rl)]\n",
    "#     # else:\n",
    "#     #     fdf = sus_df[(sus_df[\"l\"] <= rl) & (sus_df[\"r\"] >= rr)]\n",
    "    \n",
    "#     peaks.append(len(df[((df[\"l\"] >= rl) & (df[\"r\"] <= rr)) & (df[\"l\"] > df[\"r\"])]))\n",
    "#     signal.append(len(df[(df[\"l\"] > rr) & (df[\"r\"] > rl)]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import numpy as np\n",
    "# signal = np.asarray(signal)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# mutlengths = []\n",
    "# num_reads_by_cov = []\n",
    "# num_copies = []\n",
    "# signal = []\n",
    "# peaks = []\n",
    "# locs = []\n",
    "# fnames = []\n",
    "# for i in range(100):\n",
    "#     fp, aug_fp, tm, root, celldict, focal = m.genstep(seq_dict, _seqdir, log = log,\n",
    "#                                                    easy = True, only_insertions=True)\n",
    "#     is_insertion = (tm.num_copies != 0)\n",
    "#     length = tm.length\n",
    "#     rl = tm.refstart \n",
    "#     rr = rl + length\n",
    "\n",
    "#     mutlengths.append(length)\n",
    "#     locs.append((rl, rr))\n",
    "#     num_copies.append(tm.num_copies)\n",
    "#     fnames.append(aug_fp)\n",
    "\n",
    "#     num_reads_by_coverage_mult = coverage * tm.length * tm.num_copies // 300 \n",
    "#     read_dict[\"numreads\"] = num_reads_by_coverage_mult\n",
    "#     num_reads_by_cov.append(num_reads_by_coverage_mult)\n",
    "\n",
    "#     readdir = f\"{_readdir}/b{i}\"\n",
    "#     subprocess.call([\"mkdir\", \"-p\", readdir])\n",
    "    \n",
    "#     try:\n",
    "#         df, sus_df = m.readstep(read_dict, readdir, f\"b{i}\", tm, aug_fp, log = log)\n",
    "#         signal.append(len(df[((df[\"l\"] >= rl) & (df[\"r\"] <= rr)) & (df[\"l\"] > df[\"r\"])]))\n",
    "#         peaks.append(len(df[df[\"l\"] > df[\"r\"]]))\n",
    "#     except:\n",
    "#         peaks.append(-1)\n",
    "#         signal.append(-1)\n",
    "    \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample_df = df.sample(n = 500)\n",
    "# plt.scatter(mapscore_df[\"l\"], mapscore_df[\"r\"], c = mapscore_df[\"mapq\"])\n",
    "# plt.axvspan(rl, rr, alpha=0.2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import matplotlib.pyplot as plt\n",
    "# plt.boxplot(signal)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# plt.scatter(mutlengths, signal, color = \"red\")\n",
    "# plt.scatter(mutlengths, peaks, color = \"blue\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import matplotlib.pyplot as plt\n",
    "# from collections import defaultdict\n",
    "\n",
    "# # Example data\n",
    "# cats = num_copies\n",
    "# vals = signal\n",
    "\n",
    "\n",
    "# # Group valid values by category\n",
    "# cat_to_vals = defaultdict(list)\n",
    "# for c, v in zip(cats, vals):\n",
    "#     if v != -1:\n",
    "#         cat_to_vals[c].append(v)\n",
    "\n",
    "# # Sort categories for consistent x-axis order\n",
    "# sorted_cats = sorted(cat_to_vals)\n",
    "# data = [cat_to_vals[c] for c in sorted_cats]\n",
    "\n",
    "# # Plot\n",
    "# plt.boxplot(data, positions=sorted_cats)\n",
    "# plt.xticks(sorted_cats)\n",
    "# plt.xlabel(\"Copy Number\")\n",
    "# plt.ylabel(\"Signals\")\n",
    "# plt.title(\"Valid Signals by Copy Number\")\n",
    "# plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# x = [i for i in range(len(signal)) if signal[i] not in [0, -1]]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "np.float64(9593.0)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "read_dict[\"numreads\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.1', allele=0, num_copies=2, relstart=9627840, refstart=14334720, length=291840, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.0.9617280.10221120 length: 603840\n",
      "[dwgsim_core] 1 sequences, total length: 603840\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1945\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3642 sequences (364200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1750)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 149, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (148.37, 49.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 3642 reads in 0.345 CPU sec, 0.346 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b0/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b0/filtered_read2.fastq\n",
      "[main] Real time: 0.362 sec; CPU: 0.362 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=0, num_copies=2, relstart=9580800, refstart=9580800, length=158400, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.9570240.9907200 length: 336960\n",
      "[dwgsim_core] 1 sequences, total length: 336960\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1056\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1998 sequences (199800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 951)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.26, 49.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 1998 reads in 0.165 CPU sec, 0.166 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b1/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b1/filtered_read2.fastq\n",
      "[main] Real time: 0.180 sec; CPU: 0.181 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.0', allele=1, num_copies=2, relstart=17326080, refstart=15451200, length=67200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.1.17315520.17470080 length: 154560\n",
      "[dwgsim_core] 1 sequences, total length: 154560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 448\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 844 sequences (84400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 406)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 327)\n",
      "[M::mem_pestat] mean and std.dev: (151.00, 49.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 398)\n",
      "[M::mem_process_seqs] Processed 844 reads in 0.097 CPU sec, 0.108 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b2/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b2/filtered_read2.fastq\n",
      "[main] Real time: 0.123 sec; CPU: 0.113 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.1', allele=0, num_copies=2, relstart=18548160, refstart=24818880, length=6720, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor8.0.18537600.18571200 length: 33600\n",
      "[dwgsim_core] 1 sequences, total length: 33600\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 44\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 80 sequences (8000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 39)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (126, 149, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 300)\n",
      "[M::mem_pestat] mean and std.dev: (152.92, 40.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 358)\n",
      "[M::mem_process_seqs] Processed 80 reads in 0.007 CPU sec, 0.007 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b3/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b3/filtered_read2.fastq\n",
      "[main] Real time: 0.020 sec; CPU: 0.021 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell4.2', allele=0, num_copies=2, relstart=43360320, refstart=42825600, length=216960, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read2.fastq\n",
      "[main] Real time: 0.013 sec; CPU: 0.014 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "[sam_read1] reference 'ID:bwa\tPN:bwa\tVN:0.7.18-r1243-dirty\tCL:bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read2.fastq\n",
      "' is recognized as '*'.\n",
      "[main_samview] truncated file.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor7.0', allele=1, num_copies=2, relstart=8248320, refstart=8248320, length=304320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor7.1.8237760.8866560 length: 628800\n",
      "[dwgsim_core] 1 sequences, total length: 628800\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2028\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3856 sequences (385600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 946)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 146, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (148.03, 50.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 3856 reads in 0.825 CPU sec, 0.836 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b5/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b5/filtered_read2.fastq\n",
      "[main] Real time: 0.852 sec; CPU: 0.843 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.3', allele=0, num_copies=2, relstart=43214400, refstart=15282240, length=369600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.0.43203840.43963200 length: 759360\n",
      "[dwgsim_core] 1 sequences, total length: 759360\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2464\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 4684 sequences (468400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2147)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 148, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 319)\n",
      "[M::mem_pestat] mean and std.dev: (149.06, 49.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 387)\n",
      "[M::mem_process_seqs] Processed 4684 reads in 0.478 CPU sec, 0.479 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read2.fastq\n",
      "[main] Real time: 0.496 sec; CPU: 0.496 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.0', allele=0, num_copies=2, relstart=74808960, refstart=21240000, length=202560, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.0.74798400.75223680 length: 425280\n",
      "[dwgsim_core] 1 sequences, total length: 425280\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1350\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2560 sequences (256000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1265)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 180)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 306)\n",
      "[M::mem_pestat] mean and std.dev: (149.09, 47.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 369)\n",
      "[M::mem_process_seqs] Processed 2560 reads in 0.206 CPU sec, 0.206 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b7/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b7/filtered_read2.fastq\n",
      "[main] Real time: 0.221 sec; CPU: 0.222 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.1', allele=0, num_copies=3, relstart=44157120, refstart=44156160, length=540480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.0.44146560.45788160 length: 1641600\n",
      "[dwgsim_core] 1 sequences, total length: 1641600\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5404\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 10256 sequences (1025600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4690)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 312)\n",
      "[M::mem_pestat] mean and std.dev: (150.19, 49.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 377)\n",
      "[M::mem_process_seqs] Processed 10256 reads in 0.960 CPU sec, 0.962 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b8/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b8/filtered_read2.fastq\n",
      "[main] Real time: 0.984 sec; CPU: 0.983 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.1', allele=0, num_copies=2, relstart=45403200, refstart=27835200, length=467520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.0.45392640.46347840 length: 955200\n",
      "[dwgsim_core] 1 sequences, total length: 955200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3116\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 5888 sequences (588800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2904)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (150.38, 49.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 5888 reads in 0.478 CPU sec, 0.478 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b9/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b9/filtered_read2.fastq\n",
      "[main] Real time: 0.497 sec; CPU: 0.497 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.0', allele=1, num_copies=2, relstart=11383680, refstart=11383680, length=121920, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.1.11373120.11637120 length: 264000\n",
      "[dwgsim_core] 1 sequences, total length: 264000\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 812\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1538 sequences (153800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 577)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 146, 179)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 307)\n",
      "[M::mem_pestat] mean and std.dev: (147.90, 47.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 371)\n",
      "[M::mem_process_seqs] Processed 1538 reads in 0.214 CPU sec, 0.214 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b10/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b10/filtered_read2.fastq\n",
      "[main] Real time: 0.228 sec; CPU: 0.229 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.1', allele=1, num_copies=2, relstart=54463680, refstart=26237760, length=295680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.1.54453120.55064640 length: 611520\n",
      "[dwgsim_core] 1 sequences, total length: 611520\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1971\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3716 sequences (371600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1828)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 149, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 312)\n",
      "[M::mem_pestat] mean and std.dev: (150.19, 48.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 377)\n",
      "[M::mem_process_seqs] Processed 3716 reads in 0.334 CPU sec, 0.335 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b11/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b11/filtered_read2.fastq\n",
      "[main] Real time: 0.350 sec; CPU: 0.351 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.1', allele=1, num_copies=3, relstart=35392320, refstart=32750400, length=203520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.1.35381760.36012480 length: 630720\n",
      "[dwgsim_core] 1 sequences, total length: 630720\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2035\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3866 sequences (386600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1888)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.89, 50.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 3866 reads in 0.374 CPU sec, 0.375 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b12/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b12/filtered_read2.fastq\n",
      "[main] Real time: 0.391 sec; CPU: 0.392 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.0', allele=0, num_copies=2, relstart=10043520, refstart=8507520, length=157440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.0.10032960.10368000 length: 335040\n",
      "[dwgsim_core] 1 sequences, total length: 335040\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1049\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1982 sequences (198200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 237)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (120, 151, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 309)\n",
      "[M::mem_pestat] mean and std.dev: (151.73, 49.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 372)\n",
      "[M::mem_process_seqs] Processed 1982 reads in 0.298 CPU sec, 0.298 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b13/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b13/filtered_read2.fastq\n",
      "[main] Real time: 0.313 sec; CPU: 0.314 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.1', allele=1, num_copies=5, relstart=7334400, refstart=7334400, length=23040, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.1.7323840.7459200 length: 135360\n",
      "[dwgsim_core] 1 sequences, total length: 135360\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 384\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 722 sequences (72200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 359)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 151, 179)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 309)\n",
      "[M::mem_pestat] mean and std.dev: (146.52, 48.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 374)\n",
      "[M::mem_process_seqs] Processed 722 reads in 0.056 CPU sec, 0.056 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b14/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b14/filtered_read2.fastq\n",
      "[main] Real time: 0.070 sec; CPU: 0.071 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.0', allele=1, num_copies=2, relstart=54367680, refstart=32741760, length=475200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.1.54357120.55327680 length: 970560\n",
      "[dwgsim_core] 1 sequences, total length: 970560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3168\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 6014 sequences (601400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2957)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 324)\n",
      "[M::mem_pestat] mean and std.dev: (151.60, 50.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 393)\n",
      "[M::mem_process_seqs] Processed 6014 reads in 0.523 CPU sec, 0.524 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read2.fastq\n",
      "[main] Real time: 0.543 sec; CPU: 0.543 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor4.0', allele=0, num_copies=3, relstart=23601600, refstart=23600640, length=86400, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor4.0.23591040.23870400 length: 279360\n",
      "[dwgsim_core] 1 sequences, total length: 279360\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 864\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1616 sequences (161600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 805)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 153, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 322)\n",
      "[M::mem_pestat] mean and std.dev: (153.70, 48.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 1616 reads in 0.127 CPU sec, 0.127 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read2.fastq\n",
      "[main] Real time: 0.144 sec; CPU: 0.145 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.0', allele=1, num_copies=5, relstart=53541120, refstart=43128000, length=32640, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor1.1.53530560.53713920 length: 183360\n",
      "[dwgsim_core] 1 sequences, total length: 183360\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 544\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1038 sequences (103800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 507)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 152, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (149.48, 48.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 1038 reads in 0.089 CPU sec, 0.089 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b17/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b17/filtered_read2.fastq\n",
      "[main] Real time: 0.104 sec; CPU: 0.104 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.0', allele=1, num_copies=2, relstart=35324160, refstart=35323200, length=1526400, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.1.35313600.38386560 length: 3072960\n",
      "[dwgsim_core] 1 sequences, total length: 3072960\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 10176\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 19310 sequences (1931000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 9350)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.80, 49.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 19310 reads in 1.709 CPU sec, 1.714 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b18/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b18/filtered_read2.fastq\n",
      "[main] Real time: 1.779 sec; CPU: 1.751 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.0', allele=1, num_copies=3, relstart=15067200, refstart=13554240, length=83520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor8.1.15056640.15327360 length: 270720\n",
      "[dwgsim_core] 1 sequences, total length: 270720\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 835\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1578 sequences (157800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 751)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (113, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 329)\n",
      "[M::mem_pestat] mean and std.dev: (148.42, 51.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 401)\n",
      "[M::mem_process_seqs] Processed 1578 reads in 0.153 CPU sec, 0.154 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b19/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b19/filtered_read2.fastq\n",
      "[main] Real time: 0.168 sec; CPU: 0.169 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell6.1', allele=0, num_copies=3, relstart=42776640, refstart=24656640, length=26880, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.0.42766080.42866880 length: 100800\n",
      "[dwgsim_core] 1 sequences, total length: 100800\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 268\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 514 sequences (51400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 245)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 152, 187)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 341)\n",
      "[M::mem_pestat] mean and std.dev: (150.43, 55.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 418)\n",
      "[M::mem_process_seqs] Processed 514 reads in 0.050 CPU sec, 0.050 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b20/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b20/filtered_read2.fastq\n",
      "[main] Real time: 0.064 sec; CPU: 0.065 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell8.0', allele=1, num_copies=3, relstart=3327360, refstart=3327360, length=558720, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell8.1.3316800.5013120 length: 1696320\n",
      "[dwgsim_core] 1 sequences, total length: 1696320\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5587\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 10552 sequences (1055200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 10552 reads in 4.397 CPU sec, 4.414 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b21/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b21/filtered_read2.fastq\n",
      "[main] Real time: 4.434 sec; CPU: 4.418 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor4.0', allele=1, num_copies=2, relstart=29853120, refstart=29852160, length=21120, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor4.1.29842560.29904960 length: 62400\n",
      "[dwgsim_core] 1 sequences, total length: 62400\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 140\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 260 sequences (26000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 127)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (126, 157, 188)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (2, 312)\n",
      "[M::mem_pestat] mean and std.dev: (153.64, 49.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 374)\n",
      "[M::mem_process_seqs] Processed 260 reads in 0.027 CPU sec, 0.027 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b22/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b22/filtered_read2.fastq\n",
      "[main] Real time: 0.041 sec; CPU: 0.041 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.1', allele=1, num_copies=2, relstart=22444800, refstart=10101120, length=146880, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.1.22434240.22748160 length: 313920\n",
      "[dwgsim_core] 1 sequences, total length: 313920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 979\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1862 sequences (186200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 911)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 153, 187)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (152.27, 49.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 391)\n",
      "[M::mem_process_seqs] Processed 1862 reads in 0.150 CPU sec, 0.151 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b23/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b23/filtered_read2.fastq\n",
      "[main] Real time: 0.165 sec; CPU: 0.166 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.0', allele=1, num_copies=2, relstart=54034560, refstart=36644160, length=13440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.1.54024000.54071040 length: 47040\n",
      "[dwgsim_core] 1 sequences, total length: 47040\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 89\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 174 sequences (17400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 87)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (127, 153, 189)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (3, 313)\n",
      "[M::mem_pestat] mean and std.dev: (156.16, 51.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 375)\n",
      "[M::mem_process_seqs] Processed 174 reads in 0.012 CPU sec, 0.012 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b24/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b24/filtered_read2.fastq\n",
      "[main] Real time: 0.025 sec; CPU: 0.026 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.1', allele=1, num_copies=4, relstart=28771200, refstart=29504640, length=64320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.1.28760640.29038080 length: 277440\n",
      "[dwgsim_core] 1 sequences, total length: 277440\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 857\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1632 sequences (163200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 809)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (120, 152, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 312)\n",
      "[M::mem_pestat] mean and std.dev: (151.95, 48.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 376)\n",
      "[M::mem_process_seqs] Processed 1632 reads in 0.145 CPU sec, 0.145 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b25/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b25/filtered_read2.fastq\n",
      "[main] Real time: 0.160 sec; CPU: 0.160 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.0', allele=1, num_copies=3, relstart=17290560, refstart=15687360, length=247680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.1.17280000.18043200 length: 763200\n",
      "[dwgsim_core] 1 sequences, total length: 763200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2476\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 4662 sequences (466200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2143)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (150.63, 50.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 4662 reads in 0.620 CPU sec, 0.624 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b26/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b26/filtered_read2.fastq\n",
      "[main] Real time: 0.641 sec; CPU: 0.638 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.1', allele=1, num_copies=2, relstart=28968000, refstart=25090560, length=1022400, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.1.28957440.31022400 length: 2064960\n",
      "[dwgsim_core] 1 sequences, total length: 2064960\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 6816\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 12936 sequences (1293600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 6283)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.24, 49.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 12936 reads in 1.103 CPU sec, 1.104 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b27/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b27/filtered_read2.fastq\n",
      "[main] Real time: 1.127 sec; CPU: 1.125 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell6.2', allele=1, num_copies=2, relstart=10376640, refstart=24200640, length=232320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.1.10366080.10850880 length: 484800\n",
      "[dwgsim_core] 1 sequences, total length: 484800\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1548\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2944 sequences (294400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1415)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 152, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (151.30, 50.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 2944 reads in 0.301 CPU sec, 0.301 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b28/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b28/filtered_read2.fastq\n",
      "[main] Real time: 0.317 sec; CPU: 0.317 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.0', allele=1, num_copies=2, relstart=34379520, refstart=39685440, length=43200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.1.34368960.34475520 length: 106560\n",
      "[dwgsim_core] 1 sequences, total length: 106560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 288\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 544 sequences (54400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 265)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 177)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 299)\n",
      "[M::mem_pestat] mean and std.dev: (146.78, 47.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 360)\n",
      "[M::mem_process_seqs] Processed 544 reads in 0.043 CPU sec, 0.043 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b29/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b29/filtered_read2.fastq\n",
      "[main] Real time: 0.056 sec; CPU: 0.057 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell8.0', allele=1, num_copies=2, relstart=96325440, refstart=26268480, length=966720, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell8.1.96314880.98268480 length: 1953600\n",
      "[dwgsim_core] 1 sequences, total length: 1953600\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 6444\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 12228 sequences (1222800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5375)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (151.08, 49.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 12228 reads in 0.994 CPU sec, 0.995 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b30/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b30/filtered_read2.fastq\n",
      "[main] Real time: 1.017 sec; CPU: 1.017 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.0', allele=1, num_copies=2, relstart=30511680, refstart=18672960, length=416640, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.1.30501120.31354560 length: 853440\n",
      "[dwgsim_core] 1 sequences, total length: 853440\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2777\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 5272 sequences (527200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2599)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (149.93, 49.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 5272 reads in 0.415 CPU sec, 0.416 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b31/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b31/filtered_read2.fastq\n",
      "[main] Real time: 0.433 sec; CPU: 0.433 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.1', allele=0, num_copies=2, relstart=27395520, refstart=11825280, length=232320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.0.27384960.27869760 length: 484800\n",
      "[dwgsim_core] 1 sequences, total length: 484800\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1548\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2950 sequences (295000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1402)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 149, 181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 307)\n",
      "[M::mem_pestat] mean and std.dev: (148.59, 46.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 370)\n",
      "[M::mem_process_seqs] Processed 2950 reads in 0.287 CPU sec, 0.287 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b32/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b32/filtered_read2.fastq\n",
      "[main] Real time: 0.304 sec; CPU: 0.305 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.1', allele=0, num_copies=2, relstart=16202880, refstart=39004800, length=194880, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.0.16192320.16602240 length: 409920\n",
      "[dwgsim_core] 1 sequences, total length: 409920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1299\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2480 sequences (248000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1216)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 151, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (151.33, 49.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 387)\n",
      "[M::mem_process_seqs] Processed 2480 reads in 0.190 CPU sec, 0.191 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b33/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b33/filtered_read2.fastq\n",
      "[main] Real time: 0.207 sec; CPU: 0.208 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.0', allele=1, num_copies=3, relstart=31843200, refstart=30567360, length=138240, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.1.31832640.32267520 length: 434880\n",
      "[dwgsim_core] 1 sequences, total length: 434880\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1382\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2622 sequences (262200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1276)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 316)\n",
      "[M::mem_pestat] mean and std.dev: (150.85, 48.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 382)\n",
      "[M::mem_process_seqs] Processed 2622 reads in 0.259 CPU sec, 0.260 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b34/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b34/filtered_read2.fastq\n",
      "[main] Real time: 0.275 sec; CPU: 0.276 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.0', allele=0, num_copies=2, relstart=101161920, refstart=35032320, length=33600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read2.fastq\n",
      "[main] Real time: 0.013 sec; CPU: 0.014 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "[sam_read1] reference 'ID:bwa\tPN:bwa\tVN:0.7.18-r1243-dirty\tCL:bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read2.fastq\n",
      "' is recognized as '*'.\n",
      "[main_samview] truncated file.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.0', allele=0, num_copies=3, relstart=12591360, refstart=20880000, length=476160, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.0.12580800.14029440 length: 1448640\n",
      "[dwgsim_core] 1 sequences, total length: 1448640\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4761\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 9030 sequences (903000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4304)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 319)\n",
      "[M::mem_pestat] mean and std.dev: (149.84, 50.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 387)\n",
      "[M::mem_process_seqs] Processed 9030 reads in 0.838 CPU sec, 0.838 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b36/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b36/filtered_read2.fastq\n",
      "[main] Real time: 0.859 sec; CPU: 0.859 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.4', allele=0, num_copies=4, relstart=35136000, refstart=40393920, length=234240, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor8.0.35125440.36082560 length: 957120\n",
      "[dwgsim_core] 1 sequences, total length: 957120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3123\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 5968 sequences (596800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2915)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.46, 49.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 5968 reads in 0.466 CPU sec, 0.467 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b37/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b37/filtered_read2.fastq\n",
      "[main] Real time: 0.485 sec; CPU: 0.485 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.4', allele=0, num_copies=2, relstart=29867520, refstart=39019200, length=1190400, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.0.29856960.32257920 length: 2400960\n",
      "[dwgsim_core] 1 sequences, total length: 2400960\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 7936\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 15038 sequences (1503800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1620)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.96, 49.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 15038 reads in 23.344 CPU sec, 23.399 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b38/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b38/filtered_read2.fastq\n",
      "[main] Real time: 23.424 sec; CPU: 23.369 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.1', allele=0, num_copies=2, relstart=24452160, refstart=22749120, length=39360, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.0.24441600.24540480 length: 98880\n",
      "[dwgsim_core] 1 sequences, total length: 98880\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 262\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 508 sequences (50800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 254)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (128, 162, 190)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (4, 314)\n",
      "[M::mem_pestat] mean and std.dev: (158.29, 49.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 376)\n",
      "[M::mem_process_seqs] Processed 508 reads in 0.035 CPU sec, 0.035 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b39/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b39/filtered_read2.fastq\n",
      "[main] Real time: 0.049 sec; CPU: 0.050 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.3', allele=0, num_copies=2, relstart=42827520, refstart=37161600, length=1306560, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.0.42816960.45450240 length: 2633280\n",
      "[dwgsim_core] 1 sequences, total length: 2633280\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8710\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 16534 sequences (1653400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 8018)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.15, 49.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 16534 reads in 1.459 CPU sec, 1.461 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b40/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b40/filtered_read2.fastq\n",
      "[main] Real time: 1.487 sec; CPU: 1.485 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor4.0', allele=1, num_copies=2, relstart=19439040, refstart=19439040, length=275520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor4.1.19428480.19999680 length: 571200\n",
      "[dwgsim_core] 1 sequences, total length: 571200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1836\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3494 sequences (349400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1726)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 154, 187)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 325)\n",
      "[M::mem_pestat] mean and std.dev: (152.33, 50.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 394)\n",
      "[M::mem_process_seqs] Processed 3494 reads in 0.280 CPU sec, 0.280 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read2.fastq\n",
      "[main] Real time: 0.296 sec; CPU: 0.296 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.2', allele=1, num_copies=2, relstart=17637120, refstart=21874560, length=235200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.1.17626560.18117120 length: 490560\n",
      "[dwgsim_core] 1 sequences, total length: 490560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1568\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2968 sequences (296800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1455)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 149, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (148.79, 49.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 2968 reads in 0.236 CPU sec, 0.236 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b42/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b42/filtered_read2.fastq\n",
      "[main] Real time: 0.252 sec; CPU: 0.252 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.1', allele=1, num_copies=2, relstart=44860800, refstart=35963520, length=204480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.1.44850240.45279360 length: 429120\n",
      "[dwgsim_core] 1 sequences, total length: 429120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1363\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2612 sequences (261200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1238)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 149, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 322)\n",
      "[M::mem_pestat] mean and std.dev: (149.45, 50.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 391)\n",
      "[M::mem_process_seqs] Processed 2612 reads in 0.209 CPU sec, 0.209 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b43/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b43/filtered_read2.fastq\n",
      "[main] Real time: 0.225 sec; CPU: 0.226 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.2', allele=1, num_copies=3, relstart=42782400, refstart=35207040, length=293760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.1.42771840.43673280 length: 901440\n",
      "[dwgsim_core] 1 sequences, total length: 901440\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2937\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 5570 sequences (557000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2688)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 148, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (149.68, 49.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 5570 reads in 0.570 CPU sec, 0.571 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b44/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b44/filtered_read2.fastq\n",
      "[main] Real time: 0.589 sec; CPU: 0.589 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.3', allele=1, num_copies=2, relstart=38468160, refstart=32050560, length=171840, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.1.38457600.38821440 length: 363840\n",
      "[dwgsim_core] 1 sequences, total length: 363840\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1145\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2162 sequences (216200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1055)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (150.30, 49.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 392)\n",
      "[M::mem_process_seqs] Processed 2162 reads in 0.228 CPU sec, 0.229 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b45/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b45/filtered_read2.fastq\n",
      "[main] Real time: 0.243 sec; CPU: 0.244 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.2', allele=1, num_copies=2, relstart=38981760, refstart=42414720, length=26880, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.1.38971200.39045120 length: 27322\n",
      "[dwgsim_core] 1 sequences, total length: 27322\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 179\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 344 sequences (34400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 124)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (122, 157, 188)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (151.37, 50.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 386)\n",
      "[M::mem_process_seqs] Processed 344 reads in 0.103 CPU sec, 0.103 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b46/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b46/filtered_read2.fastq\n",
      "[main] Real time: 0.117 sec; CPU: 0.118 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.1', allele=1, num_copies=2, relstart=9408000, refstart=9408000, length=911040, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.1.9397440.11239680 length: 1842240\n",
      "[dwgsim_core] 1 sequences, total length: 1842240\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 6073\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 11558 sequences (1155800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3160)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 325)\n",
      "[M::mem_pestat] mean and std.dev: (149.88, 49.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 395)\n",
      "[M::mem_process_seqs] Processed 11558 reads in 1.822 CPU sec, 1.824 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b47/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b47/filtered_read2.fastq\n",
      "[main] Real time: 1.844 sec; CPU: 1.843 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.2', allele=0, num_copies=2, relstart=52135680, refstart=9405120, length=91200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.0.52125120.52327680 length: 202560\n",
      "[dwgsim_core] 1 sequences, total length: 202560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 608\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1170 sequences (117000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 253)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 152, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (151.45, 51.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 1170 reads in 0.305 CPU sec, 0.306 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read2.fastq\n",
      "[main] Real time: 0.319 sec; CPU: 0.320 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.0', allele=1, num_copies=2, relstart=32195520, refstart=32194560, length=156480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.1.32184960.32518080 length: 333120\n",
      "[dwgsim_core] 1 sequences, total length: 333120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1043\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1974 sequences (197400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 938)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 149, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (150.55, 48.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 1974 reads in 0.198 CPU sec, 0.199 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b49/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b49/filtered_read2.fastq\n",
      "[main] Real time: 0.214 sec; CPU: 0.215 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.0', allele=1, num_copies=2, relstart=9882240, refstart=9882240, length=48000, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.1.9871680.9987840 length: 116160\n",
      "[dwgsim_core] 1 sequences, total length: 116160\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1040\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2004 sequences (200400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (102, 135, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 261)\n",
      "[M::mem_pestat] mean and std.dev: (129.10, 42.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 314)\n",
      "[M::mem_process_seqs] Processed 2004 reads in 0.222 CPU sec, 0.222 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b0/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b0/filtered_read2.fastq\n",
      "[main] Real time: 0.238 sec; CPU: 0.239 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.3', allele=1, num_copies=2, relstart=32299200, refstart=26552640, length=66240, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.1.32288640.32441280 length: 152640\n",
      "[dwgsim_core] 1 sequences, total length: 152640\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1435\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2778 sequences (277800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1379)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 311)\n",
      "[M::mem_pestat] mean and std.dev: (149.56, 49.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 376)\n",
      "[M::mem_process_seqs] Processed 2778 reads in 0.194 CPU sec, 0.194 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b1/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b1/filtered_read2.fastq\n",
      "[main] Real time: 0.210 sec; CPU: 0.210 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell6.2', allele=0, num_copies=2, relstart=6216000, refstart=15782400, length=509760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.0.6205440.7245120 length: 1039680\n",
      "[dwgsim_core] 1 sequences, total length: 1039680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 11044\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 20986 sequences (2098600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 9946)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.91, 49.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 20986 reads in 2.182 CPU sec, 2.183 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b2/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b2/filtered_read2.fastq\n",
      "[main] Real time: 2.212 sec; CPU: 2.210 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.1', allele=1, num_copies=4, relstart=72926400, refstart=44680320, length=154560, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.1.72915840.73554240 length: 638400\n",
      "[dwgsim_core] 1 sequences, total length: 638400\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 6697\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 12728 sequences (1272800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5904)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.52, 49.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 12728 reads in 1.288 CPU sec, 1.289 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b3/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b3/filtered_read2.fastq\n",
      "[main] Real time: 1.311 sec; CPU: 1.310 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.1', allele=1, num_copies=2, relstart=82945920, refstart=20149440, length=183360, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.1.82935360.83322240 length: 386880\n",
      "[dwgsim_core] 1 sequences, total length: 386880\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3972\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 7504 sequences (750400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3639)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.00, 49.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 7504 reads in 0.622 CPU sec, 0.622 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read2.fastq\n",
      "[main] Real time: 0.643 sec; CPU: 0.643 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.0', allele=0, num_copies=2, relstart=10669440, refstart=11705280, length=1793280, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.0.10658880.14265600 length: 3606720\n",
      "[dwgsim_core] 1 sequences, total length: 3606720\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 38854\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 73822 sequences (7382200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 33725)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.92, 49.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 73822 reads in 6.891 CPU sec, 6.912 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b5/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b5/filtered_read2.fastq\n",
      "[main] Real time: 6.975 sec; CPU: 6.955 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor7.0', allele=1, num_copies=2, relstart=2805120, refstart=2805120, length=625920, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor7.1.2794560.4066560 length: 1272000\n",
      "[dwgsim_core] 1 sequences, total length: 1272000\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 13561\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 25812 sequences (2581200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4954)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (113, 149, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (148.55, 49.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 25812 reads in 13.304 CPU sec, 13.330 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read2.fastq\n",
      "[main] Real time: 13.360 sec; CPU: 13.336 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.1', allele=1, num_copies=3, relstart=74354880, refstart=32665920, length=93120, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.1.74344320.74643840 length: 299520\n",
      "[dwgsim_core] 1 sequences, total length: 299520\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3026\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 5700 sequences (570000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2803)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (149.74, 49.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 5700 reads in 0.449 CPU sec, 0.450 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b7/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b7/filtered_read2.fastq\n",
      "[main] Real time: 0.469 sec; CPU: 0.469 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor7.0', allele=1, num_copies=2, relstart=16657920, refstart=5775360, length=29760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor7.1.16647360.16727040 length: 79680\n",
      "[dwgsim_core] 1 sequences, total length: 79680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 644\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1216 sequences (121600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 1216 reads in 0.404 CPU sec, 0.422 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b8/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b8/filtered_read2.fastq\n",
      "[main] Real time: 0.436 sec; CPU: 0.436 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.0', allele=0, num_copies=2, relstart=2929920, refstart=2929920, length=237120, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.0.2919360.3413760 length: 494400\n",
      "[dwgsim_core] 1 sequences, total length: 494400\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5137\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 9782 sequences (978200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3159)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 150, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 310)\n",
      "[M::mem_pestat] mean and std.dev: (150.22, 48.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 374)\n",
      "[M::mem_process_seqs] Processed 9782 reads in 2.936 CPU sec, 2.939 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b9/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b9/filtered_read2.fastq\n",
      "[main] Real time: 2.960 sec; CPU: 2.957 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.1', allele=0, num_copies=4, relstart=30252480, refstart=30307200, length=44160, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.0.30241920.30438720 length: 196800\n",
      "[dwgsim_core] 1 sequences, total length: 196800\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1913\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3628 sequences (362800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1774)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 151, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 322)\n",
      "[M::mem_pestat] mean and std.dev: (150.85, 49.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 3628 reads in 0.326 CPU sec, 0.326 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b10/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b10/filtered_read2.fastq\n",
      "[main] Real time: 0.343 sec; CPU: 0.344 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell6.0', allele=1, num_copies=2, relstart=83159040, refstart=40184640, length=220800, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.1.83148480.83610240 length: 461760\n",
      "[dwgsim_core] 1 sequences, total length: 461760\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4784\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 9094 sequences (909400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4430)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 316)\n",
      "[M::mem_pestat] mean and std.dev: (151.43, 48.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 382)\n",
      "[M::mem_process_seqs] Processed 9094 reads in 0.739 CPU sec, 0.740 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b11/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b11/filtered_read2.fastq\n",
      "[main] Real time: 0.761 sec; CPU: 0.761 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.0', allele=1, num_copies=2, relstart=40080000, refstart=40079040, length=203520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.1.40069440.40496640 length: 427200\n",
      "[dwgsim_core] 1 sequences, total length: 427200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4409\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 8378 sequences (837800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3860)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 148, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (148.51, 49.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 8378 reads in 0.727 CPU sec, 0.728 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b12/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b12/filtered_read2.fastq\n",
      "[main] Real time: 0.747 sec; CPU: 0.748 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.0', allele=1, num_copies=5, relstart=14277120, refstart=20499840, length=16320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.1.14266560.14368320 length: 101760\n",
      "[dwgsim_core] 1 sequences, total length: 101760\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 884\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1682 sequences (168200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 836)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 148, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 314)\n",
      "[M::mem_pestat] mean and std.dev: (148.54, 49.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 380)\n",
      "[M::mem_process_seqs] Processed 1682 reads in 0.108 CPU sec, 0.108 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b13/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b13/filtered_read2.fastq\n",
      "[main] Real time: 0.122 sec; CPU: 0.123 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.4', allele=1, num_copies=2, relstart=16632000, refstart=14492160, length=709440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor1.1.16621440.18060480 length: 1439040\n",
      "[dwgsim_core] 1 sequences, total length: 1439040\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 15371\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 29290 sequences (2929000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 14086)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (149.92, 49.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 29290 reads in 2.576 CPU sec, 2.578 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b14/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b14/filtered_read2.fastq\n",
      "[main] Real time: 2.611 sec; CPU: 2.610 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.5', allele=1, num_copies=2, relstart=4015680, refstart=6376320, length=1428480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.1.4005120.6882240 length: 2877120\n",
      "[dwgsim_core] 1 sequences, total length: 2877120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 30950\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 58902 sequences (5890200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 7186)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.84, 49.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 58902 reads in 14.420 CPU sec, 14.440 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read2.fastq\n",
      "[main] Real time: 14.497 sec; CPU: 14.478 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.1', allele=0, num_copies=4, relstart=24334080, refstart=29616000, length=115200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read2.fastq\n",
      "[main] Real time: 0.012 sec; CPU: 0.014 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "[sam_read1] reference 'ID:bwa\tPN:bwa\tVN:0.7.18-r1243-dirty\tCL:bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read2.fastq\n",
      "' is recognized as '*'.\n",
      "[main_samview] truncated file.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell6.1', allele=0, num_copies=3, relstart=9725760, refstart=3598080, length=133440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.0.9715200.10135680 length: 420480\n",
      "[dwgsim_core] 1 sequences, total length: 420480\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4336\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 8258 sequences (825800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 8258 reads in 3.399 CPU sec, 3.412 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b17/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b17/filtered_read2.fastq\n",
      "[main] Real time: 3.431 sec; CPU: 3.419 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.0', allele=1, num_copies=4, relstart=16459200, refstart=16459200, length=200640, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor8.1.16448640.17271360 length: 822720\n",
      "[dwgsim_core] 1 sequences, total length: 822720\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8694\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 16542 sequences (1654200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 8111)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.51, 49.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 16542 reads in 1.317 CPU sec, 1.319 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b18/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b18/filtered_read2.fastq\n",
      "[main] Real time: 1.344 sec; CPU: 1.343 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor7.2', allele=0, num_copies=3, relstart=8914560, refstart=8914560, length=77760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor7.0.8904000.9157440 length: 253440\n",
      "[dwgsim_core] 1 sequences, total length: 253440\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2527\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 4792 sequences (479200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2280)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 152, 187)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 325)\n",
      "[M::mem_pestat] mean and std.dev: (151.57, 49.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 394)\n",
      "[M::mem_process_seqs] Processed 4792 reads in 0.771 CPU sec, 0.772 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b19/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b19/filtered_read2.fastq\n",
      "[main] Real time: 0.789 sec; CPU: 0.789 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.0', allele=1, num_copies=5, relstart=24757440, refstart=20196480, length=791040, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.1.24746880.28722240 length: 3975360\n",
      "[dwgsim_core] 1 sequences, total length: 3975360\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 42848\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 81380 sequences (8138000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 39722)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.80, 49.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 81380 reads in 6.831 CPU sec, 6.839 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b20/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b20/filtered_read2.fastq\n",
      "[main] Real time: 6.918 sec; CPU: 6.905 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.1', allele=0, num_copies=2, relstart=32759040, refstart=41031360, length=763200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.0.32748480.34295040 length: 1546560\n",
      "[dwgsim_core] 1 sequences, total length: 1546560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 16536\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 31334 sequences (3133400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 14852)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.97, 49.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 31334 reads in 2.943 CPU sec, 2.945 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b21/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b21/filtered_read2.fastq\n",
      "[main] Real time: 2.980 sec; CPU: 2.978 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor4.1', allele=0, num_copies=2, relstart=64335360, refstart=40015680, length=67200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor4.0.64324800.64479360 length: 154560\n",
      "[dwgsim_core] 1 sequences, total length: 154560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1456\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2760 sequences (276000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1356)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 149, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (149.32, 50.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 2760 reads in 0.221 CPU sec, 0.221 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b22/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b22/filtered_read2.fastq\n",
      "[main] Real time: 0.237 sec; CPU: 0.237 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.2', allele=1, num_copies=2, relstart=8023680, refstart=8023680, length=103680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.1.8013120.8240640 length: 227520\n",
      "[dwgsim_core] 1 sequences, total length: 227520\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2246\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 4276 sequences (427600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1597)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 148, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.40, 51.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 4276 reads in 0.521 CPU sec, 0.522 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b23/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b23/filtered_read2.fastq\n",
      "[main] Real time: 0.539 sec; CPU: 0.539 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.0', allele=0, num_copies=2, relstart=42245760, refstart=42216960, length=445440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.0.42235200.43146240 length: 911040\n",
      "[dwgsim_core] 1 sequences, total length: 911040\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 9651\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 18280 sequences (1828000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 8936)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.63, 49.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 18280 reads in 1.484 CPU sec, 1.498 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b24/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b24/filtered_read2.fastq\n",
      "[main] Real time: 1.526 sec; CPU: 1.512 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor7.0', allele=0, num_copies=4, relstart=6740160, refstart=11471040, length=203520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor7.0.6729600.7563840 length: 834240\n",
      "[dwgsim_core] 1 sequences, total length: 834240\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8819\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 16766 sequences (1676600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2227)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 149, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 322)\n",
      "[M::mem_pestat] mean and std.dev: (149.34, 50.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 391)\n",
      "[M::mem_process_seqs] Processed 16766 reads in 1.941 CPU sec, 1.943 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b25/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b25/filtered_read2.fastq\n",
      "[main] Real time: 1.969 sec; CPU: 1.967 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.3', allele=0, num_copies=2, relstart=6558720, refstart=11065920, length=528000, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.0.6548160.7624320 length: 1076160\n",
      "[dwgsim_core] 1 sequences, total length: 1076160\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 11440\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 21698 sequences (2169800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3898)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 151, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 322)\n",
      "[M::mem_pestat] mean and std.dev: (150.98, 49.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 21698 reads in 22.020 CPU sec, 22.054 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b26/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b26/filtered_read2.fastq\n",
      "[main] Real time: 22.083 sec; CPU: 22.049 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.2', allele=0, num_copies=3, relstart=7091520, refstart=7091520, length=271680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.7080960.7916160 length: 835200\n",
      "[dwgsim_core] 1 sequences, total length: 835200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8829\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 16798 sequences (1679800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 3834)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 149, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (148.89, 50.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 386)\n",
      "[M::mem_process_seqs] Processed 16798 reads in 4.915 CPU sec, 4.921 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b27/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b27/filtered_read2.fastq\n",
      "[main] Real time: 4.947 sec; CPU: 4.941 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=0, num_copies=2, relstart=34755840, refstart=34754880, length=492480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.34745280.35750400 length: 1005120\n",
      "[dwgsim_core] 1 sequences, total length: 1005120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 10670\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 20236 sequences (2023600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 9873)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.79, 49.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 20236 reads in 1.680 CPU sec, 1.681 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b28/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b28/filtered_read2.fastq\n",
      "[main] Real time: 1.709 sec; CPU: 1.708 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.0', allele=1, num_copies=2, relstart=19815360, refstart=16370880, length=923520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.1.19804800.21672000 length: 1867200\n",
      "[dwgsim_core] 1 sequences, total length: 1867200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 20009\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 37974 sequences (3797400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 18367)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.65, 49.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 37974 reads in 3.028 CPU sec, 3.030 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b29/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b29/filtered_read2.fastq\n",
      "[main] Real time: 3.069 sec; CPU: 3.068 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.2', allele=0, num_copies=2, relstart=27049920, refstart=27048960, length=72000, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.0.27039360.27203520 length: 164160\n",
      "[dwgsim_core] 1 sequences, total length: 164160\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1560\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2974 sequences (297400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1421)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 149, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (150.83, 50.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 2974 reads in 0.265 CPU sec, 0.265 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b30/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b30/filtered_read2.fastq\n",
      "[main] Real time: 0.282 sec; CPU: 0.282 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell8.4', allele=0, num_copies=3, relstart=61606080, refstart=44040000, length=57600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell8.0.61595520.61788480 length: 192960\n",
      "[dwgsim_core] 1 sequences, total length: 192960\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1872\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3566 sequences (356600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1705)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 151, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 313)\n",
      "[M::mem_pestat] mean and std.dev: (150.27, 47.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 378)\n",
      "[M::mem_process_seqs] Processed 3566 reads in 0.292 CPU sec, 0.292 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b31/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b31/filtered_read2.fastq\n",
      "[main] Real time: 0.333 sec; CPU: 0.310 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.2', allele=0, num_copies=3, relstart=49383360, refstart=38066880, length=247680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.0.49372800.50136000 length: 763200\n",
      "[dwgsim_core] 1 sequences, total length: 763200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8049\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 15312 sequences (1531200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 7486)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.38, 49.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 15312 reads in 1.297 CPU sec, 1.308 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b32/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b32/filtered_read2.fastq\n",
      "[main] Real time: 1.331 sec; CPU: 1.321 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell6.1', allele=0, num_copies=3, relstart=30032640, refstart=9420480, length=63360, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.0.30022080.30232320 length: 210240\n",
      "[dwgsim_core] 1 sequences, total length: 210240\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2059\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3904 sequences (390400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 901)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 149, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 322)\n",
      "[M::mem_pestat] mean and std.dev: (150.27, 52.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 391)\n",
      "[M::mem_process_seqs] Processed 3904 reads in 0.938 CPU sec, 0.940 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b33/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b33/filtered_read2.fastq\n",
      "[main] Real time: 0.958 sec; CPU: 0.957 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.0', allele=1, num_copies=2, relstart=10772160, refstart=10772160, length=75840, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.1.10761600.10933440 length: 171840\n",
      "[dwgsim_core] 1 sequences, total length: 171840\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1643\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3118 sequences (311800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 95)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (104, 138, 177)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (142.91, 50.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 396)\n",
      "[M::mem_process_seqs] Processed 3118 reads in 0.954 CPU sec, 0.955 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b34/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b34/filtered_read2.fastq\n",
      "[main] Real time: 0.972 sec; CPU: 0.971 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.1', allele=0, num_copies=4, relstart=41720640, refstart=36203520, length=9600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.0.41710080.41768640 length: 58560\n",
      "[dwgsim_core] 1 sequences, total length: 58560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 416\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 792 sequences (79200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 388)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (120, 143, 180)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 300)\n",
      "[M::mem_pestat] mean and std.dev: (147.37, 47.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 360)\n",
      "[M::mem_process_seqs] Processed 792 reads in 0.065 CPU sec, 0.065 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read2.fastq\n",
      "[main] Real time: 0.079 sec; CPU: 0.080 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.0', allele=0, num_copies=3, relstart=23793600, refstart=15514560, length=13440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.0.23783040.23843520 length: 60480\n",
      "[dwgsim_core] 1 sequences, total length: 60480\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 436\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 838 sequences (83800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 265)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 150, 181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (148.89, 50.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 382)\n",
      "[M::mem_process_seqs] Processed 838 reads in 0.186 CPU sec, 0.187 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b36/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b36/filtered_read2.fastq\n",
      "[main] Real time: 0.201 sec; CPU: 0.201 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor4.0', allele=0, num_copies=2, relstart=26656320, refstart=27581760, length=485760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor4.0.26645760.27637440 length: 991680\n",
      "[dwgsim_core] 1 sequences, total length: 991680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 10524\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 20066 sequences (2006600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 9773)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (149.85, 49.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 20066 reads in 1.737 CPU sec, 1.738 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b37/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b37/filtered_read2.fastq\n",
      "[main] Real time: 1.767 sec; CPU: 1.765 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell6.1', allele=1, num_copies=4, relstart=23606400, refstart=23605440, length=273600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.1.23595840.24710400 length: 1114560\n",
      "[dwgsim_core] 1 sequences, total length: 1114560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 11856\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 22544 sequences (2254400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 10980)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (151.06, 48.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 22544 reads in 1.824 CPU sec, 1.826 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b38/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b38/filtered_read2.fastq\n",
      "[main] Real time: 1.855 sec; CPU: 1.853 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell8.0', allele=0, num_copies=2, relstart=42111360, refstart=30756480, length=78720, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell8.0.42100800.42278400 length: 177600\n",
      "[dwgsim_core] 1 sequences, total length: 177600\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1705\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3228 sequences (322800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1575)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (151.03, 50.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 392)\n",
      "[M::mem_process_seqs] Processed 3228 reads in 0.325 CPU sec, 0.325 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b39/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b39/filtered_read2.fastq\n",
      "[main] Real time: 0.341 sec; CPU: 0.342 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.0', allele=1, num_copies=2, relstart=33846720, refstart=33845760, length=702720, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor1.1.33836160.35261760 length: 1425600\n",
      "[dwgsim_core] 1 sequences, total length: 1425600\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 15225\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 29004 sequences (2900400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 14043)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 151, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 313)\n",
      "[M::mem_pestat] mean and std.dev: (150.32, 48.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 378)\n",
      "[M::mem_process_seqs] Processed 29004 reads in 2.688 CPU sec, 2.695 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b40/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b40/filtered_read2.fastq\n",
      "[main] Real time: 2.728 sec; CPU: 2.723 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.1', allele=0, num_copies=5, relstart=48939840, refstart=45034560, length=33600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read2.fastq\n",
      "[main] Real time: 0.013 sec; CPU: 0.015 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "[sam_read1] reference 'ID:bwa\tPN:bwa\tVN:0.7.18-r1243-dirty\tCL:bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read2.fastq\n",
      "' is recognized as '*'.\n",
      "[main_samview] truncated file.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell8.1', allele=1, num_copies=2, relstart=15016320, refstart=8424000, length=300480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell8.1.15005760.15626880 length: 621120\n",
      "[dwgsim_core] 1 sequences, total length: 621120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 6510\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 12380 sequences (1238000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2317)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 150, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 316)\n",
      "[M::mem_pestat] mean and std.dev: (148.67, 50.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 383)\n",
      "[M::mem_process_seqs] Processed 12380 reads in 1.538 CPU sec, 2.037 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b42/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b42/filtered_read2.fastq\n",
      "[main] Real time: 2.060 sec; CPU: 2.040 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor4.3', allele=0, num_copies=7, relstart=2168640, refstart=2168640, length=34560, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor4.0.2158080.2420160 length: 262080\n",
      "[dwgsim_core] 1 sequences, total length: 262080\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2620\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 4954 sequences (495400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 193)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 150, 190)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 342)\n",
      "[M::mem_pestat] mean and std.dev: (147.73, 54.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 418)\n",
      "[M::mem_process_seqs] Processed 4954 reads in 0.733 CPU sec, 0.734 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b43/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b43/filtered_read2.fastq\n",
      "[main] Real time: 0.751 sec; CPU: 0.752 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.2', allele=0, num_copies=5, relstart=8732160, refstart=15612480, length=169920, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.0.8721600.9591360 length: 869760\n",
      "[dwgsim_core] 1 sequences, total length: 869760\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 9204\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 17530 sequences (1753000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 7596)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.01, 49.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 17530 reads in 2.215 CPU sec, 2.217 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b44/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b44/filtered_read2.fastq\n",
      "[main] Real time: 2.243 sec; CPU: 2.241 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell4.1', allele=1, num_copies=4, relstart=7190400, refstart=15144960, length=559680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell4.1.7179840.9438720 length: 2258880\n",
      "[dwgsim_core] 1 sequences, total length: 2258880\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 24252\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 46092 sequences (4609200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 21175)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.36, 49.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 46092 reads in 4.478 CPU sec, 4.483 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b45/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b45/filtered_read2.fastq\n",
      "[main] Real time: 4.529 sec; CPU: 4.524 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.2', allele=0, num_copies=2, relstart=37938240, refstart=39805440, length=973440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.0.37927680.39894720 length: 1967040\n",
      "[dwgsim_core] 1 sequences, total length: 1967040\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 21091\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 40124 sequences (4012400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 19605)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.43, 49.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 40124 reads in 3.235 CPU sec, 3.255 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b46/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b46/filtered_read2.fastq\n",
      "[main] Real time: 3.297 sec; CPU: 3.278 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.0', allele=0, num_copies=3, relstart=5380800, refstart=5380800, length=259200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor1.0.5370240.6168000 length: 797760\n",
      "[dwgsim_core] 1 sequences, total length: 797760\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8424\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 16048 sequences (1604800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 463)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 311)\n",
      "[M::mem_pestat] mean and std.dev: (148.51, 48.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 376)\n",
      "[M::mem_process_seqs] Processed 16048 reads in 13.188 CPU sec, 13.209 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b47/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b47/filtered_read2.fastq\n",
      "[main] Real time: 13.234 sec; CPU: 13.214 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.0', allele=0, num_copies=3, relstart=36066240, refstart=39941760, length=134400, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read2.fastq\n",
      "[main] Real time: 0.013 sec; CPU: 0.015 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "[sam_read1] reference 'ID:bwa\tPN:bwa\tVN:0.7.18-r1243-dirty\tCL:bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read2.fastq\n",
      "' is recognized as '*'.\n",
      "[main_samview] truncated file.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.0', allele=1, num_copies=3, relstart=2543040, refstart=2543040, length=760320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.1.2532480.4833600 length: 2301120\n",
      "[dwgsim_core] 1 sequences, total length: 2301120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 24710\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 46926 sequences (4692600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 12753)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.48, 49.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 46926 reads in 15.066 CPU sec, 15.143 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b49/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b49/filtered_read2.fastq\n",
      "[main] Real time: 15.190 sec; CPU: 15.114 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=0, num_copies=2, relstart=590400, refstart=590400, length=36480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.579840.672960 length: 93120\n",
      "[dwgsim_core] 1 sequences, total length: 93120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1337\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2536 sequences (253600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 2536 reads in 1.209 CPU sec, 1.211 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b0/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b0/filtered_read2.fastq\n",
      "[main] Real time: 1.225 sec; CPU: 1.225 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor6.2', allele=0, num_copies=2, relstart=8009280, refstart=12230400, length=160320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor6.0.7998720.8339520 length: 340800\n",
      "[dwgsim_core] 1 sequences, total length: 340800\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5878\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 11214 sequences (1121400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4038)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.08, 49.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 11214 reads in 1.078 CPU sec, 1.079 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b1/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b1/filtered_read2.fastq\n",
      "[main] Real time: 1.102 sec; CPU: 1.101 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=1, num_copies=3, relstart=27604800, refstart=29498880, length=329280, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.1.27594240.28602240 length: 1008000\n",
      "[dwgsim_core] 1 sequences, total length: 1008000\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 18110\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 34328 sequences (3432800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 16881)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.06, 49.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 34328 reads in 3.210 CPU sec, 3.213 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b2/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b2/filtered_read2.fastq\n",
      "[main] Real time: 3.251 sec; CPU: 3.249 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.3', allele=0, num_copies=2, relstart=3380160, refstart=8616960, length=651840, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.0.3369600.4693440 length: 1323840\n",
      "[dwgsim_core] 1 sequences, total length: 1323840\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 23900\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 45414 sequences (4541400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 18556)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (149.92, 49.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 45414 reads in 6.212 CPU sec, 6.228 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b3/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b3/filtered_read2.fastq\n",
      "[main] Real time: 6.273 sec; CPU: 6.258 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.1', allele=0, num_copies=3, relstart=37925760, refstart=38128320, length=361920, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.37915200.39021120 length: 1105920\n",
      "[dwgsim_core] 1 sequences, total length: 1105920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 19905\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 37782 sequences (3778200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 18475)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.20, 49.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 37782 reads in 3.288 CPU sec, 3.291 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read2.fastq\n",
      "[main] Real time: 3.331 sec; CPU: 3.328 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=0, num_copies=4, relstart=11151360, refstart=11151360, length=69120, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.11140800.11437440 length: 296640\n",
      "[dwgsim_core] 1 sequences, total length: 296640\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5068\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 9596 sequences (959600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 9596 reads in 10.607 CPU sec, 10.621 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b5/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b5/filtered_read2.fastq\n",
      "[main] Real time: 10.641 sec; CPU: 10.628 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.1', allele=0, num_copies=2, relstart=45530880, refstart=15720000, length=666240, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.0.45520320.46872960 length: 1352640\n",
      "[dwgsim_core] 1 sequences, total length: 1352640\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 24428\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 46288 sequences (4628800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 20759)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (150.29, 49.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 46288 reads in 4.777 CPU sec, 4.781 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read2.fastq\n",
      "[main] Real time: 4.826 sec; CPU: 4.823 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.3', allele=1, num_copies=2, relstart=31685760, refstart=41498880, length=386880, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.1.31675200.32469120 length: 793920\n",
      "[dwgsim_core] 1 sequences, total length: 793920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 14185\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 26860 sequences (2686000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 12613)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 319)\n",
      "[M::mem_pestat] mean and std.dev: (151.21, 49.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 386)\n",
      "[M::mem_process_seqs] Processed 26860 reads in 2.456 CPU sec, 2.458 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b7/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b7/filtered_read2.fastq\n",
      "[main] Real time: 2.489 sec; CPU: 2.488 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor7.0', allele=1, num_copies=2, relstart=15079680, refstart=15079680, length=936960, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor7.1.15069120.16963200 length: 1894080\n",
      "[dwgsim_core] 1 sequences, total length: 1894080\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 34355\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 65292 sequences (6529200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 30216)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.80, 49.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 65292 reads in 6.830 CPU sec, 6.843 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b8/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b8/filtered_read2.fastq\n",
      "[main] Real time: 6.904 sec; CPU: 6.890 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.0', allele=0, num_copies=7, relstart=19352640, refstart=19352640, length=36480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.0.19342080.19617600 length: 275520\n",
      "[dwgsim_core] 1 sequences, total length: 275520\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4681\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 8870 sequences (887000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4387)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.10, 49.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 8870 reads in 0.697 CPU sec, 0.698 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b9/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b9/filtered_read2.fastq\n",
      "[main] Real time: 0.720 sec; CPU: 0.720 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor6.2', allele=1, num_copies=2, relstart=20691840, refstart=33801600, length=1181760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor6.1.20681280.23064960 length: 2383680\n",
      "[dwgsim_core] 1 sequences, total length: 2383680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 43331\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 82206 sequences (8220600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 39898)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.07, 49.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 82206 reads in 7.965 CPU sec, 7.972 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b10/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b10/filtered_read2.fastq\n",
      "[main] Real time: 8.045 sec; CPU: 8.038 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.0', allele=1, num_copies=3, relstart=38603520, refstart=30310080, length=8640, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.1.38592960.38639040 length: 46080\n",
      "[dwgsim_core] 1 sequences, total length: 46080\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 475\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 898 sequences (89800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 447)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 187)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 327)\n",
      "[M::mem_pestat] mean and std.dev: (151.21, 51.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 397)\n",
      "[M::mem_process_seqs] Processed 898 reads in 0.060 CPU sec, 0.060 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b11/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b11/filtered_read2.fastq\n",
      "[main] Real time: 0.074 sec; CPU: 0.075 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.4', allele=0, num_copies=2, relstart=20586240, refstart=21330240, length=581760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.0.20575680.21759360 length: 1183680\n",
      "[dwgsim_core] 1 sequences, total length: 1183680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 21331\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 40508 sequences (4050800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 19488)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.25, 49.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 40508 reads in 3.734 CPU sec, 3.737 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b12/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b12/filtered_read2.fastq\n",
      "[main] Real time: 3.779 sec; CPU: 3.776 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=0, num_copies=2, relstart=17430720, refstart=17430720, length=138240, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.17420160.17716800 length: 296640\n",
      "[dwgsim_core] 1 sequences, total length: 296640\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5068\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 9638 sequences (963800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4482)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 319)\n",
      "[M::mem_pestat] mean and std.dev: (149.11, 50.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 387)\n",
      "[M::mem_process_seqs] Processed 9638 reads in 0.990 CPU sec, 0.991 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b13/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b13/filtered_read2.fastq\n",
      "[main] Real time: 1.011 sec; CPU: 1.011 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.1', allele=0, num_copies=2, relstart=40918080, refstart=39995520, length=278400, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.0.40907520.41484480 length: 576960\n",
      "[dwgsim_core] 1 sequences, total length: 576960\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 10208\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 19374 sequences (1937400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 9438)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (150.42, 49.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 19374 reads in 1.491 CPU sec, 1.493 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b14/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b14/filtered_read2.fastq\n",
      "[main] Real time: 1.521 sec; CPU: 1.519 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=1, num_copies=2, relstart=34991040, refstart=34990080, length=262080, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.1.34980480.35524800 length: 544320\n",
      "[dwgsim_core] 1 sequences, total length: 544320\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 9609\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 18314 sequences (1831400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 8869)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 314)\n",
      "[M::mem_pestat] mean and std.dev: (149.68, 49.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 380)\n",
      "[M::mem_process_seqs] Processed 18314 reads in 1.746 CPU sec, 1.758 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read2.fastq\n",
      "[main] Real time: 1.783 sec; CPU: 1.772 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.1', allele=0, num_copies=2, relstart=12793920, refstart=13345920, length=265920, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.0.12783360.13335360 length: 552000\n",
      "[dwgsim_core] 1 sequences, total length: 552000\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 9750\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 18544 sequences (1854400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4060)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 319)\n",
      "[M::mem_pestat] mean and std.dev: (150.86, 49.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 386)\n",
      "[M::mem_process_seqs] Processed 18544 reads in 3.516 CPU sec, 3.520 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read2.fastq\n",
      "[main] Real time: 3.548 sec; CPU: 3.544 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.1', allele=1, num_copies=3, relstart=1882560, refstart=1882560, length=713280, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.1.1872000.4032000 length: 2160000\n",
      "[dwgsim_core] 1 sequences, total length: 2160000\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 39230\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 74466 sequences (7446600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 18326)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.38, 49.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 74466 reads in 33.439 CPU sec, 34.765 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b17/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b17/filtered_read2.fastq\n",
      "[main] Real time: 34.835 sec; CPU: 33.509 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell8.1', allele=1, num_copies=4, relstart=36086400, refstart=16799040, length=48960, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell8.1.36075840.36291840 length: 216000\n",
      "[dwgsim_core] 1 sequences, total length: 216000\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3590\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 6808 sequences (680800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3209)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 152, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (151.31, 49.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 6808 reads in 0.703 CPU sec, 0.708 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b18/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b18/filtered_read2.fastq\n",
      "[main] Real time: 0.740 sec; CPU: 0.730 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.1', allele=0, num_copies=2, relstart=5872320, refstart=5872320, length=95040, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.0.5861760.6072000 length: 210240\n",
      "[dwgsim_core] 1 sequences, total length: 210240\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3484\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 6642 sequences (664200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3088)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 152, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (151.43, 50.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 6642 reads in 1.175 CPU sec, 1.177 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b19/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b19/filtered_read2.fastq\n",
      "[main] Real time: 1.200 sec; CPU: 1.194 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor6.0', allele=0, num_copies=2, relstart=17336640, refstart=11339520, length=116160, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor6.0.17326080.17578560 length: 252480\n",
      "[dwgsim_core] 1 sequences, total length: 252480\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4259\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 8072 sequences (807200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3838)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.83, 49.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 8072 reads in 0.880 CPU sec, 0.881 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b20/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b20/filtered_read2.fastq\n",
      "[main] Real time: 0.901 sec; CPU: 0.902 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell4.1', allele=0, num_copies=2, relstart=30829440, refstart=35860800, length=417600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell4.0.30818880.31674240 length: 855360\n",
      "[dwgsim_core] 1 sequences, total length: 855360\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 15312\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 28990 sequences (2899000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 14033)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.88, 49.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 28990 reads in 2.403 CPU sec, 2.423 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b21/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b21/filtered_read2.fastq\n",
      "[main] Real time: 2.457 sec; CPU: 2.438 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.1', allele=0, num_copies=2, relstart=3819840, refstart=14071680, length=48960, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.0.3809280.3927360 length: 118080\n",
      "[dwgsim_core] 1 sequences, total length: 118080\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1795\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3418 sequences (341800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1699)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 152, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (151.09, 50.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 392)\n",
      "[M::mem_process_seqs] Processed 3418 reads in 0.255 CPU sec, 0.255 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b22/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b22/filtered_read2.fastq\n",
      "[main] Real time: 0.271 sec; CPU: 0.272 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=1, num_copies=3, relstart=26345280, refstart=26344320, length=53760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.1.26334720.26516160 length: 181440\n",
      "[dwgsim_core] 1 sequences, total length: 181440\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2956\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 5614 sequences (561400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2669)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 148, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 325)\n",
      "[M::mem_pestat] mean and std.dev: (149.67, 50.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 395)\n",
      "[M::mem_process_seqs] Processed 5614 reads in 0.522 CPU sec, 0.523 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b23/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b23/filtered_read2.fastq\n",
      "[main] Real time: 0.541 sec; CPU: 0.542 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.2', allele=0, num_copies=3, relstart=43817280, refstart=42023040, length=520320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.43806720.45387840 length: 1581120\n",
      "[dwgsim_core] 1 sequences, total length: 1581120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 28617\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 54376 sequences (5437600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 26321)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.49, 49.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 54376 reads in 5.085 CPU sec, 5.089 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b24/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b24/filtered_read2.fastq\n",
      "[main] Real time: 5.142 sec; CPU: 5.138 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.0', allele=1, num_copies=2, relstart=25300800, refstart=20344320, length=12480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor8.1.25290240.25335360 length: 45120\n",
      "[dwgsim_core] 1 sequences, total length: 45120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 457\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 866 sequences (86600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 426)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (120, 155, 187)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (152.74, 49.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 866 reads in 0.078 CPU sec, 0.078 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b25/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b25/filtered_read2.fastq\n",
      "[main] Real time: 0.092 sec; CPU: 0.093 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor6.0', allele=0, num_copies=2, relstart=1220160, refstart=1220160, length=454080, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor6.0.1209600.2137920 length: 928320\n",
      "[dwgsim_core] 1 sequences, total length: 928320\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 16649\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 31666 sequences (3166600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 175)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 145, 174)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 292)\n",
      "[M::mem_pestat] mean and std.dev: (143.47, 50.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 351)\n",
      "[M::mem_process_seqs] Processed 31666 reads in 76.008 CPU sec, 76.144 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b26/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b26/filtered_read2.fastq\n",
      "[main] Real time: 76.180 sec; CPU: 76.045 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.0', allele=0, num_copies=2, relstart=59871360, refstart=36586560, length=272640, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.0.59860800.60426240 length: 565440\n",
      "[dwgsim_core] 1 sequences, total length: 565440\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 9996\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 18986 sequences (1898600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 9367)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.35, 49.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 18986 reads in 1.405 CPU sec, 1.420 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b27/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b27/filtered_read2.fastq\n",
      "[main] Real time: 1.447 sec; CPU: 1.432 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.1', allele=0, num_copies=2, relstart=15831360, refstart=13915200, length=133440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.0.15820800.16107840 length: 287040\n",
      "[dwgsim_core] 1 sequences, total length: 287040\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4892\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 9282 sequences (928200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4516)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 152, 187)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 327)\n",
      "[M::mem_pestat] mean and std.dev: (151.74, 49.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 397)\n",
      "[M::mem_process_seqs] Processed 9282 reads in 0.791 CPU sec, 0.792 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b28/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b28/filtered_read2.fastq\n",
      "[main] Real time: 0.813 sec; CPU: 0.813 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor4.0', allele=1, num_copies=2, relstart=2759040, refstart=2759040, length=177600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor4.1.2748480.3123840 length: 375360\n",
      "[dwgsim_core] 1 sequences, total length: 375360\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 6512\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 12326 sequences (1232600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 4400)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 316)\n",
      "[M::mem_pestat] mean and std.dev: (150.24, 48.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 382)\n",
      "[M::mem_process_seqs] Processed 12326 reads in 1.240 CPU sec, 1.241 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b29/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b29/filtered_read2.fastq\n",
      "[main] Real time: 1.263 sec; CPU: 1.262 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.0', allele=1, num_copies=2, relstart=18429120, refstart=18429120, length=116160, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.1.18418560.18671040 length: 252480\n",
      "[dwgsim_core] 1 sequences, total length: 252480\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4259\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 8128 sequences (812800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3950)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (150.34, 49.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 8128 reads in 0.627 CPU sec, 0.627 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b30/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b30/filtered_read2.fastq\n",
      "[main] Real time: 0.649 sec; CPU: 0.649 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.0', allele=0, num_copies=2, relstart=5385600, refstart=5888640, length=619200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.0.5375040.6633600 length: 1258560\n",
      "[dwgsim_core] 1 sequences, total length: 1258560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 22704\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 43144 sequences (4314400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 14152)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.33, 49.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 43144 reads in 8.336 CPU sec, 8.367 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b31/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b31/filtered_read2.fastq\n",
      "[main] Real time: 8.413 sec; CPU: 8.381 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.1', allele=0, num_copies=2, relstart=14526720, refstart=14526720, length=10560, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.0.14516160.14557440 length: 41280\n",
      "[dwgsim_core] 1 sequences, total length: 41280\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 387\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 736 sequences (73600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 362)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 319)\n",
      "[M::mem_pestat] mean and std.dev: (152.15, 49.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 386)\n",
      "[M::mem_process_seqs] Processed 736 reads in 0.060 CPU sec, 0.060 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b32/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b32/filtered_read2.fastq\n",
      "[main] Real time: 0.074 sec; CPU: 0.075 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.1', allele=1, num_copies=2, relstart=40041600, refstart=31874880, length=220800, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.1.40031040.40492800 length: 461760\n",
      "[dwgsim_core] 1 sequences, total length: 461760\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8096\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 15406 sequences (1540600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 7605)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.20, 49.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 15406 reads in 1.416 CPU sec, 1.417 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b33/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b33/filtered_read2.fastq\n",
      "[main] Real time: 1.443 sec; CPU: 1.441 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor4.0', allele=1, num_copies=2, relstart=30346560, refstart=28666560, length=433920, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor4.1.30336000.31224000 length: 888000\n",
      "[dwgsim_core] 1 sequences, total length: 888000\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 15910\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 30184 sequences (3018400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 14296)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (150.12, 49.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 30184 reads in 2.953 CPU sec, 2.956 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b34/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b34/filtered_read2.fastq\n",
      "[main] Real time: 2.990 sec; CPU: 2.989 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.0', allele=1, num_copies=3, relstart=41928000, refstart=19806720, length=552000, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.1.41917440.43593600 length: 1676160\n",
      "[dwgsim_core] 1 sequences, total length: 1676160\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 30360\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 57724 sequences (5772400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 28167)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.07, 49.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 57724 reads in 4.744 CPU sec, 4.749 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read2.fastq\n",
      "[main] Real time: 4.803 sec; CPU: 4.799 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.1', allele=1, num_copies=3, relstart=28556160, refstart=34272000, length=32640, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.1.28545600.28663680 length: 118080\n",
      "[dwgsim_core] 1 sequences, total length: 118080\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1795\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3386 sequences (338600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1593)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 311)\n",
      "[M::mem_pestat] mean and std.dev: (149.50, 48.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 376)\n",
      "[M::mem_process_seqs] Processed 3386 reads in 0.273 CPU sec, 0.273 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b36/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b36/filtered_read2.fastq\n",
      "[main] Real time: 0.289 sec; CPU: 0.289 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=0, num_copies=3, relstart=29537280, refstart=24205440, length=369600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.29526720.30655680 length: 1128960\n",
      "[dwgsim_core] 1 sequences, total length: 1128960\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 20328\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 38640 sequences (3864000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 18655)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.41, 49.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 38640 reads in 3.913 CPU sec, 3.932 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b37/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b37/filtered_read2.fastq\n",
      "[main] Real time: 3.980 sec; CPU: 3.957 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.3', allele=0, num_copies=2, relstart=85348800, refstart=40751040, length=84480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.0.85338240.85527360 length: 189120\n",
      "[dwgsim_core] 1 sequences, total length: 189120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3097\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 5874 sequences (587400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2868)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 152, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 313)\n",
      "[M::mem_pestat] mean and std.dev: (150.57, 49.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 378)\n",
      "[M::mem_process_seqs] Processed 5874 reads in 0.551 CPU sec, 0.552 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b38/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b38/filtered_read2.fastq\n",
      "[main] Real time: 0.573 sec; CPU: 0.572 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.0', allele=1, num_copies=3, relstart=26698560, refstart=21252480, length=75840, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.1.26688000.26935680 length: 247680\n",
      "[dwgsim_core] 1 sequences, total length: 247680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4171\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 7872 sequences (787200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3576)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.68, 49.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 7872 reads in 0.873 CPU sec, 0.874 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b39/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b39/filtered_read2.fastq\n",
      "[main] Real time: 0.897 sec; CPU: 0.896 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.0', allele=1, num_copies=2, relstart=24853440, refstart=31553280, length=333120, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.1.24842880.25529280 length: 686400\n",
      "[dwgsim_core] 1 sequences, total length: 686400\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 12214\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 23222 sequences (2322200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 10996)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.27, 49.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 23222 reads in 2.635 CPU sec, 2.637 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b40/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b40/filtered_read2.fastq\n",
      "[main] Real time: 2.667 sec; CPU: 2.666 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.0', allele=0, num_copies=2, relstart=2340480, refstart=2340480, length=7680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.0.2329920.2365440 length: 35520\n",
      "[dwgsim_core] 1 sequences, total length: 35520\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 281\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 534 sequences (53400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 24)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (144, 158, 202)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (28, 318)\n",
      "[M::mem_pestat] mean and std.dev: (168.46, 42.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 376)\n",
      "[M::mem_process_seqs] Processed 534 reads in 0.128 CPU sec, 0.128 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read2.fastq\n",
      "[main] Real time: 0.142 sec; CPU: 0.143 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.0', allele=0, num_copies=2, relstart=5404800, refstart=5404800, length=561600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.0.5394240.6537600 length: 1143360\n",
      "[dwgsim_core] 1 sequences, total length: 1143360\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 20592\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 39102 sequences (3910200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 6720)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.82, 49.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 39102 reads in 17.465 CPU sec, 17.493 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b42/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b42/filtered_read2.fastq\n",
      "[main] Real time: 17.537 sec; CPU: 17.509 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell4.2', allele=1, num_copies=2, relstart=44192640, refstart=20175360, length=639360, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell4.1.44182080.45480960 length: 1298880\n",
      "[dwgsim_core] 1 sequences, total length: 1298880\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 23443\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 44610 sequences (4461000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 21818)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.16, 49.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 44610 reads in 3.556 CPU sec, 3.559 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b43/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b43/filtered_read2.fastq\n",
      "[main] Real time: 3.617 sec; CPU: 3.611 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.2', allele=0, num_copies=3, relstart=31245120, refstart=10024320, length=75840, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor8.0.31234560.31482240 length: 247680\n",
      "[dwgsim_core] 1 sequences, total length: 247680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4171\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 7944 sequences (794400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2174)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 324)\n",
      "[M::mem_pestat] mean and std.dev: (149.70, 50.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 394)\n",
      "[M::mem_process_seqs] Processed 7944 reads in 1.459 CPU sec, 1.461 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b44/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b44/filtered_read2.fastq\n",
      "[main] Real time: 1.483 sec; CPU: 1.481 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.1', allele=1, num_copies=3, relstart=31137600, refstart=33816000, length=114240, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.1.31127040.31489920 length: 362880\n",
      "[dwgsim_core] 1 sequences, total length: 362880\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 6283\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 11948 sequences (1194800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5558)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (150.71, 49.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 11948 reads in 1.228 CPU sec, 1.230 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b45/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b45/filtered_read2.fastq\n",
      "[main] Real time: 1.253 sec; CPU: 1.252 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.1', allele=0, num_copies=2, relstart=25197120, refstart=25196160, length=578880, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.0.25186560.26364480 length: 1177920\n",
      "[dwgsim_core] 1 sequences, total length: 1177920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 21225\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 40322 sequences (4032200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 19782)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.15, 49.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 40322 reads in 3.145 CPU sec, 3.163 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b46/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b46/filtered_read2.fastq\n",
      "[main] Real time: 3.210 sec; CPU: 3.191 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell6.1', allele=1, num_copies=2, relstart=23037120, refstart=7750080, length=119040, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.1.23026560.23284800 length: 258240\n",
      "[dwgsim_core] 1 sequences, total length: 258240\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4364\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 8298 sequences (829800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 385)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 148, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 328)\n",
      "[M::mem_pestat] mean and std.dev: (149.96, 50.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_process_seqs] Processed 8298 reads in 1.769 CPU sec, 1.772 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b47/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b47/filtered_read2.fastq\n",
      "[main] Real time: 1.794 sec; CPU: 1.791 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.0', allele=0, num_copies=2, relstart=14284800, refstart=14284800, length=1043520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor8.0.14274240.16381440 length: 2107200\n",
      "[dwgsim_core] 1 sequences, total length: 2107200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 38262\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 72624 sequences (7262400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 33056)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.15, 49.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 72624 reads in 6.939 CPU sec, 6.967 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read2.fastq\n",
      "[main] Real time: 7.041 sec; CPU: 7.009 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.2', allele=1, num_copies=4, relstart=118350720, refstart=19756800, length=96960, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor8.1.118340160.118748160 length: 408000\n",
      "[dwgsim_core] 1 sequences, total length: 408000\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 7110\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 13498 sequences (1349800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 6596)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (150.61, 49.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 13498 reads in 1.106 CPU sec, 1.106 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b49/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b49/filtered_read2.fastq\n",
      "[main] Real time: 1.151 sec; CPU: 1.138 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell4.0', allele=0, num_copies=3, relstart=3194880, refstart=6626880, length=39360, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell4.0.3184320.3322560 length: 138240\n",
      "[dwgsim_core] 1 sequences, total length: 138240\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3050\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 5756 sequences (575600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 211)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 153, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 336)\n",
      "[M::mem_pestat] mean and std.dev: (147.44, 52.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 411)\n",
      "[M::mem_process_seqs] Processed 5756 reads in 1.200 CPU sec, 1.201 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b0/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b0/filtered_read2.fastq\n",
      "[main] Real time: 1.224 sec; CPU: 1.221 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.4', allele=0, num_copies=2, relstart=97914240, refstart=25936320, length=132480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.0.97903680.98188800 length: 285120\n",
      "[dwgsim_core] 1 sequences, total length: 285120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 6844\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 12996 sequences (1299600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5939)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 319)\n",
      "[M::mem_pestat] mean and std.dev: (151.27, 49.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 386)\n",
      "[M::mem_process_seqs] Processed 12996 reads in 1.396 CPU sec, 1.417 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b1/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b1/filtered_read2.fastq\n",
      "[main] Real time: 1.444 sec; CPU: 1.423 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=1, num_copies=2, relstart=1897920, refstart=1897920, length=221760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.1.1887360.2351040 length: 463680\n",
      "[dwgsim_core] 1 sequences, total length: 463680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 11457\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 21764 sequences (2176400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 6856)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.28, 49.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 21764 reads in 18.111 CPU sec, 18.150 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b2/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b2/filtered_read2.fastq\n",
      "[main] Real time: 18.180 sec; CPU: 18.141 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.1', allele=0, num_copies=2, relstart=67109760, refstart=21564480, length=185280, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor1.0.67099200.67489920 length: 390720\n",
      "[dwgsim_core] 1 sequences, total length: 390720\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 9572\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 18168 sequences (1816800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 8771)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.06, 50.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 18168 reads in 1.640 CPU sec, 1.657 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b3/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b3/filtered_read2.fastq\n",
      "[main] Real time: 1.702 sec; CPU: 1.675 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell8.0', allele=1, num_copies=3, relstart=60462720, refstart=40450560, length=57600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell8.1.60452160.60645120 length: 192960\n",
      "[dwgsim_core] 1 sequences, total length: 192960\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4464\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 8482 sequences (848200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4155)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 151, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (150.00, 49.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 8482 reads in 0.620 CPU sec, 0.620 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read2.fastq\n",
      "[main] Real time: 0.642 sec; CPU: 0.642 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.0', allele=0, num_copies=3, relstart=167232000, refstart=27860160, length=303360, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.0.167221440.168151680 length: 930240\n",
      "[dwgsim_core] 1 sequences, total length: 930240\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 23510\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 44626 sequences (4462600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 21728)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.06, 49.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 44626 reads in 4.456 CPU sec, 4.464 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b5/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b5/filtered_read2.fastq\n",
      "[main] Real time: 4.515 sec; CPU: 4.505 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.2', allele=1, num_copies=3, relstart=2570880, refstart=2570880, length=159360, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.1.2560320.3058560 length: 498240\n",
      "[dwgsim_core] 1 sequences, total length: 498240\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 12350\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 23458 sequences (2345800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 6693)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.69, 50.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 23458 reads in 2.950 CPU sec, 2.953 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read2.fastq\n",
      "[main] Real time: 2.988 sec; CPU: 2.983 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell8.2', allele=1, num_copies=2, relstart=30971520, refstart=31053120, length=101760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell8.1.30960960.31184640 length: 223680\n",
      "[dwgsim_core] 1 sequences, total length: 223680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5257\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 10030 sequences (1003000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 4901)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 152, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (151.68, 48.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 383)\n",
      "[M::mem_process_seqs] Processed 10030 reads in 0.879 CPU sec, 0.879 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b7/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b7/filtered_read2.fastq\n",
      "[main] Real time: 0.902 sec; CPU: 0.902 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor4.1', allele=0, num_copies=3, relstart=7143360, refstart=6058560, length=337920, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor4.0.7132800.8166720 length: 1033920\n",
      "[dwgsim_core] 1 sequences, total length: 1033920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 26188\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 49868 sequences (4986800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 14341)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 319)\n",
      "[M::mem_pestat] mean and std.dev: (149.35, 49.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 387)\n",
      "[M::mem_process_seqs] Processed 49868 reads in 9.019 CPU sec, 9.041 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b8/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b8/filtered_read2.fastq\n",
      "[main] Real time: 9.091 sec; CPU: 9.070 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=1, num_copies=2, relstart=28689600, refstart=30041280, length=165120, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.1.28679040.29029440 length: 350400\n",
      "[dwgsim_core] 1 sequences, total length: 350400\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8531\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 16228 sequences (1622800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 7722)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.66, 49.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 16228 reads in 1.616 CPU sec, 1.618 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b9/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b9/filtered_read2.fastq\n",
      "[main] Real time: 1.643 sec; CPU: 1.641 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.3', allele=1, num_copies=5, relstart=68376000, refstart=33500160, length=131520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.1.68365440.69043200 length: 677760\n",
      "[dwgsim_core] 1 sequences, total length: 677760\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 16988\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 32218 sequences (3221800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 15829)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.90, 49.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 32218 reads in 2.391 CPU sec, 2.393 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b10/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b10/filtered_read2.fastq\n",
      "[main] Real time: 2.429 sec; CPU: 2.427 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.2', allele=1, num_copies=3, relstart=20843520, refstart=13560000, length=353280, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.1.20832960.21912960 length: 1080000\n",
      "[dwgsim_core] 1 sequences, total length: 1080000\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 27379\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 51958 sequences (5195800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 25383)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.98, 49.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 51958 reads in 4.210 CPU sec, 4.213 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b11/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b11/filtered_read2.fastq\n",
      "[main] Real time: 4.262 sec; CPU: 4.259 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor6.0', allele=0, num_copies=3, relstart=24255360, refstart=25086720, length=268800, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor6.0.24244800.25071360 length: 826560\n",
      "[dwgsim_core] 1 sequences, total length: 826560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 20832\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 39576 sequences (3957600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 19379)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.41, 49.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 39576 reads in 3.074 CPU sec, 3.094 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b12/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b12/filtered_read2.fastq\n",
      "[main] Real time: 3.135 sec; CPU: 3.115 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.0', allele=0, num_copies=2, relstart=3532800, refstart=3532800, length=712320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.0.3522240.4967040 length: 1444800\n",
      "[dwgsim_core] 1 sequences, total length: 1444800\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 36803\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 69856 sequences (6985600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69856 reads in 29.551 CPU sec, 131.122 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b13/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b13/filtered_read2.fastq\n",
      "[main] Real time: 131.182 sec; CPU: 29.613 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.0', allele=1, num_copies=2, relstart=22554240, refstart=22553280, length=264960, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.1.22543680.23093760 length: 550080\n",
      "[dwgsim_core] 1 sequences, total length: 550080\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 13689\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 26054 sequences (2605400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 12807)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.36, 49.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 26054 reads in 2.070 CPU sec, 2.080 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b14/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b14/filtered_read2.fastq\n",
      "[main] Real time: 2.111 sec; CPU: 2.102 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.0', allele=1, num_copies=3, relstart=50011200, refstart=33113280, length=82560, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read2.fastq\n",
      "[main] Real time: 0.013 sec; CPU: 0.014 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "[sam_read1] reference 'ID:bwa\tPN:bwa\tVN:0.7.18-r1243-dirty\tCL:bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read2.fastq\n",
      "' is recognized as '*'.\n",
      "[main_samview] truncated file.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.1', allele=1, num_copies=2, relstart=20160000, refstart=26034240, length=108480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.1.20149440.20386560 length: 237120\n",
      "[dwgsim_core] 1 sequences, total length: 237120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5604\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 10636 sequences (1063600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5161)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 148, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 316)\n",
      "[M::mem_pestat] mean and std.dev: (148.84, 49.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 383)\n",
      "[M::mem_process_seqs] Processed 10636 reads in 0.925 CPU sec, 0.925 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read2.fastq\n",
      "[main] Real time: 0.948 sec; CPU: 0.949 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.0', allele=1, num_copies=2, relstart=3492480, refstart=3492480, length=933120, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.1.3481920.5368320 length: 1886400\n",
      "[dwgsim_core] 1 sequences, total length: 1886400\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 48211\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 91580 sequences (9158000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 91580 reads in 38.454 CPU sec, 45.101 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b17/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b17/filtered_read2.fastq\n",
      "[main] Real time: 45.182 sec; CPU: 38.536 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell8.0', allele=1, num_copies=2, relstart=26365440, refstart=42122880, length=701760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell8.1.26354880.27778560 length: 1423680\n",
      "[dwgsim_core] 1 sequences, total length: 1423680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 36257\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 68780 sequences (6878000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 33514)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.80, 49.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 68780 reads in 6.371 CPU sec, 6.382 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b18/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b18/filtered_read2.fastq\n",
      "[main] Real time: 6.450 sec; CPU: 6.437 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor6.1', allele=1, num_copies=2, relstart=933120, refstart=933120, length=31680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor6.1.922560.1006080 length: 83520\n",
      "[dwgsim_core] 1 sequences, total length: 83520\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1636\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3106 sequences (310600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 26)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (106, 150, 165)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 283)\n",
      "[M::mem_pestat] mean and std.dev: (137.46, 46.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 342)\n",
      "[M::mem_process_seqs] Processed 3106 reads in 2.947 CPU sec, 2.955 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b19/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b19/filtered_read2.fastq\n",
      "[main] Real time: 2.971 sec; CPU: 2.964 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.0', allele=1, num_copies=4, relstart=42806400, refstart=42805440, length=100800, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor1.1.42795840.43219200 length: 423360\n",
      "[dwgsim_core] 1 sequences, total length: 423360\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 10416\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 19896 sequences (1989600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 9328)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.69, 49.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 19896 reads in 1.588 CPU sec, 1.588 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b20/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b20/filtered_read2.fastq\n",
      "[main] Real time: 1.619 sec; CPU: 1.619 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.1', allele=0, num_copies=4, relstart=8018880, refstart=8018880, length=31680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.0.8008320.8155200 length: 146880\n",
      "[dwgsim_core] 1 sequences, total length: 146880\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3273\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 6236 sequences (623600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1864)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 150, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 308)\n",
      "[M::mem_pestat] mean and std.dev: (150.08, 49.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 371)\n",
      "[M::mem_process_seqs] Processed 6236 reads in 0.679 CPU sec, 0.680 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b21/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b21/filtered_read2.fastq\n",
      "[main] Real time: 0.700 sec; CPU: 0.699 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.0', allele=1, num_copies=3, relstart=39896640, refstart=19524480, length=653760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.1.39886080.41867520 length: 1981440\n",
      "[dwgsim_core] 1 sequences, total length: 1981440\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 50666\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 96352 sequences (9635200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 47286)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.78, 49.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 96352 reads in 8.102 CPU sec, 8.221 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b22/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b22/filtered_read2.fastq\n",
      "[main] Real time: 8.318 sec; CPU: 8.199 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.0', allele=1, num_copies=4, relstart=34057920, refstart=26150400, length=187200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.1.34047360.34816320 length: 768960\n",
      "[dwgsim_core] 1 sequences, total length: 768960\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 19344\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 36810 sequences (3681000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 15278)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.17, 49.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 36810 reads in 3.540 CPU sec, 3.568 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b23/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b23/filtered_read2.fastq\n",
      "[main] Real time: 3.616 sec; CPU: 3.586 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.2', allele=1, num_copies=2, relstart=39647040, refstart=42487680, length=340800, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.1.39636480.40338240 length: 701760\n",
      "[dwgsim_core] 1 sequences, total length: 701760\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 17608\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 33424 sequences (3342400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 15831)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.93, 49.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 33424 reads in 3.291 CPU sec, 3.309 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b24/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b24/filtered_read2.fastq\n",
      "[main] Real time: 3.354 sec; CPU: 3.333 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.1', allele=0, num_copies=2, relstart=29117760, refstart=30241920, length=112320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.0.29107200.29352000 length: 244800\n",
      "[dwgsim_core] 1 sequences, total length: 244800\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5803\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 10948 sequences (1094800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5379)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.16, 48.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 10948 reads in 1.116 CPU sec, 1.119 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b25/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b25/filtered_read2.fastq\n",
      "[main] Real time: 1.143 sec; CPU: 1.140 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.1', allele=0, num_copies=2, relstart=40155840, refstart=33873600, length=98880, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.0.40145280.40363200 length: 217920\n",
      "[dwgsim_core] 1 sequences, total length: 217920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5108\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 9664 sequences (966400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4418)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 316)\n",
      "[M::mem_pestat] mean and std.dev: (150.98, 49.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 382)\n",
      "[M::mem_process_seqs] Processed 9664 reads in 1.105 CPU sec, 1.108 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b26/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b26/filtered_read2.fastq\n",
      "[main] Real time: 1.130 sec; CPU: 1.128 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.1', allele=0, num_copies=2, relstart=79169280, refstart=34360320, length=451200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor1.0.79158720.80081280 length: 922560\n",
      "[dwgsim_core] 1 sequences, total length: 922560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 23312\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 44268 sequences (4426800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 21459)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.92, 49.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 44268 reads in 5.674 CPU sec, 5.698 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b27/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b27/filtered_read2.fastq\n",
      "[main] Real time: 5.753 sec; CPU: 5.727 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.1', allele=1, num_copies=4, relstart=16484160, refstart=4880640, length=42240, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor8.1.16473600.16662720 length: 189120\n",
      "[dwgsim_core] 1 sequences, total length: 189120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4364\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 8324 sequences (832400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 8324 reads in 3.690 CPU sec, 3.721 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b28/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b28/filtered_read2.fastq\n",
      "[main] Real time: 3.753 sec; CPU: 3.718 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.0', allele=1, num_copies=3, relstart=31564800, refstart=30728640, length=35520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.1.31554240.31680960 length: 126720\n",
      "[dwgsim_core] 1 sequences, total length: 126720\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 2752\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 5230 sequences (523000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2533)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 151, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 311)\n",
      "[M::mem_pestat] mean and std.dev: (150.52, 48.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 375)\n",
      "[M::mem_process_seqs] Processed 5230 reads in 0.500 CPU sec, 0.501 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b29/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b29/filtered_read2.fastq\n",
      "[main] Real time: 0.522 sec; CPU: 0.521 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.1', allele=0, num_copies=2, relstart=28473600, refstart=35551680, length=94080, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.0.28463040.28671360 length: 208320\n",
      "[dwgsim_core] 1 sequences, total length: 208320\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4860\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 9220 sequences (922000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4494)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 149, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.53, 48.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 9220 reads in 0.980 CPU sec, 0.983 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b30/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b30/filtered_read2.fastq\n",
      "[main] Real time: 1.004 sec; CPU: 1.002 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='founder.0', allele=1, num_copies=2, relstart=5283840, refstart=5283840, length=1148160, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] founder.1.5273280.7589760 length: 2316480\n",
      "[dwgsim_core] 1 sequences, total length: 2316480\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 59321\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 12780 sequences (1278000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 21695)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.20, 49.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 41.384 CPU sec, 42.353 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2720)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 151, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (150.00, 49.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 12780 reads in 5.309 CPU sec, 5.292 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b31/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b31/filtered_read2.fastq\n",
      "[main] Real time: 47.712 sec; CPU: 46.754 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.3', allele=0, num_copies=3, relstart=9207360, refstart=9207360, length=406080, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.0.9196800.10435200 length: 1238400\n",
      "[dwgsim_core] 1 sequences, total length: 1238400\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 31471\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 59908 sequences (5990800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 23884)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.78, 49.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 59908 reads in 8.528 CPU sec, 8.561 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b32/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b32/filtered_read2.fastq\n",
      "[main] Real time: 8.636 sec; CPU: 8.600 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.2', allele=0, num_copies=2, relstart=72947520, refstart=22703040, length=143040, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.0.72936960.73243200 length: 306240\n",
      "[dwgsim_core] 1 sequences, total length: 306240\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 7390\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 14028 sequences (1402800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 6881)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (151.39, 49.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 14028 reads in 1.215 CPU sec, 1.262 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b33/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b33/filtered_read2.fastq\n",
      "[main] Real time: 1.286 sec; CPU: 1.239 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.1', allele=1, num_copies=2, relstart=33545280, refstart=37547520, length=90240, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.1.33534720.33735360 length: 200640\n",
      "[dwgsim_core] 1 sequences, total length: 200640\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4662\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 8878 sequences (887800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4410)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (151.17, 49.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 8878 reads in 0.798 CPU sec, 0.801 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b34/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b34/filtered_read2.fastq\n",
      "[main] Real time: 0.822 sec; CPU: 0.820 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor7.0', allele=0, num_copies=3, relstart=43460160, refstart=34269120, length=99840, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor7.0.43449600.43769280 length: 319680\n",
      "[dwgsim_core] 1 sequences, total length: 319680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 7737\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 14678 sequences (1467800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 7125)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (150.77, 49.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 14678 reads in 1.407 CPU sec, 1.449 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read2.fastq\n",
      "[main] Real time: 1.474 sec; CPU: 1.433 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.0', allele=0, num_copies=2, relstart=39884160, refstart=44195520, length=404160, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.0.39873600.40702080 length: 828480\n",
      "[dwgsim_core] 1 sequences, total length: 828480\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 20881\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 39640 sequences (3964000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 19406)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.27, 49.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 39640 reads in 3.555 CPU sec, 3.568 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b36/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b36/filtered_read2.fastq\n",
      "[main] Real time: 3.610 sec; CPU: 3.597 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.0', allele=0, num_copies=3, relstart=2001600, refstart=19082880, length=79680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor8.0.1991040.2250240 length: 259200\n",
      "[dwgsim_core] 1 sequences, total length: 259200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 6175\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 11794 sequences (1179400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5839)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.60, 49.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 11794 reads in 0.955 CPU sec, 0.957 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b37/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b37/filtered_read2.fastq\n",
      "[main] Real time: 0.981 sec; CPU: 0.980 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell6.1', allele=0, num_copies=2, relstart=60978240, refstart=29205120, length=1130880, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.0.60967680.63249600 length: 2281920\n",
      "[dwgsim_core] 1 sequences, total length: 2281920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 58428\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 10866 sequences (1086600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 49014)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.15, 49.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 9.408 CPU sec, 9.448 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5312)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 149, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 325)\n",
      "[M::mem_pestat] mean and std.dev: (150.01, 50.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 395)\n",
      "[M::mem_process_seqs] Processed 10866 reads in 1.097 CPU sec, 1.056 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b38/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b38/filtered_read2.fastq\n",
      "[main] Real time: 10.558 sec; CPU: 10.556 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.0', allele=1, num_copies=3, relstart=52807680, refstart=42066240, length=2069760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.1.52797120.59026560 length: 5504122\n",
      "[dwgsim_core] 1 sequences, total length: 5504122\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 160406\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 47159)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.20, 49.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 9.273 CPU sec, 9.288 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 47157)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.17, 49.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 9.201 CPU sec, 9.172 real sec\n",
      "[M::process] read 4740 sequences (474000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 47123)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.34, 49.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 9.179 CPU sec, 9.145 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2225)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (150.93, 49.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 4740 reads in 0.486 CPU sec, 0.439 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b39/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b39/filtered_read2.fastq\n",
      "[main] Real time: 28.097 sec; CPU: 28.190 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.0', allele=0, num_copies=2, relstart=6945600, refstart=6945600, length=186240, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor1.0.6935040.7327680 length: 392640\n",
      "[dwgsim_core] 1 sequences, total length: 392640\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 9622\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 18326 sequences (1832600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4090)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 149, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 316)\n",
      "[M::mem_pestat] mean and std.dev: (148.50, 50.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 383)\n",
      "[M::mem_process_seqs] Processed 18326 reads in 2.641 CPU sec, 2.643 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b40/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b40/filtered_read2.fastq\n",
      "[main] Real time: 2.674 sec; CPU: 2.671 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.2', allele=1, num_copies=2, relstart=10727040, refstart=10727040, length=471360, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.1.10716480.11679360 length: 962880\n",
      "[dwgsim_core] 1 sequences, total length: 962880\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 24353\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 46256 sequences (4625600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 393)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (108, 147, 178)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (143.44, 50.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 46256 reads in 59.704 CPU sec, 60.306 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read2.fastq\n",
      "[main] Real time: 60.374 sec; CPU: 59.761 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor6.4', allele=0, num_copies=2, relstart=30301440, refstart=30300480, length=115200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor6.0.30290880.30541440 length: 250560\n",
      "[dwgsim_core] 1 sequences, total length: 250560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5952\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 11374 sequences (1137400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5541)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 324)\n",
      "[M::mem_pestat] mean and std.dev: (150.82, 49.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 393)\n",
      "[M::mem_process_seqs] Processed 11374 reads in 1.224 CPU sec, 1.229 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b42/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b42/filtered_read2.fastq\n",
      "[main] Real time: 1.255 sec; CPU: 1.250 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.0', allele=0, num_copies=2, relstart=22342080, refstart=24680640, length=160320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.0.22331520.22672320 length: 340800\n",
      "[dwgsim_core] 1 sequences, total length: 340800\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8283\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 15726 sequences (1572600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 7668)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (150.39, 50.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 392)\n",
      "[M::mem_process_seqs] Processed 15726 reads in 1.544 CPU sec, 1.578 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b43/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b43/filtered_read2.fastq\n",
      "[main] Real time: 1.609 sec; CPU: 1.572 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.1', allele=1, num_copies=2, relstart=6613440, refstart=6613440, length=630720, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.1.6602880.7884480 length: 1281600\n",
      "[dwgsim_core] 1 sequences, total length: 1281600\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 32587\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 62018 sequences (6201800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 17479)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.61, 49.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 62018 reads in 15.195 CPU sec, 15.279 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b44/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b44/filtered_read2.fastq\n",
      "[main] Real time: 15.343 sec; CPU: 15.259 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.0', allele=0, num_copies=5, relstart=10926720, refstart=12528000, length=28800, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.0.10916160.11080320 length: 164160\n",
      "[dwgsim_core] 1 sequences, total length: 164160\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3720\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 7068 sequences (706800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3163)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 314)\n",
      "[M::mem_pestat] mean and std.dev: (149.06, 49.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 380)\n",
      "[M::mem_process_seqs] Processed 7068 reads in 0.971 CPU sec, 0.980 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b45/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b45/filtered_read2.fastq\n",
      "[main] Real time: 1.015 sec; CPU: 0.996 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell4.0', allele=0, num_copies=2, relstart=23192640, refstart=38047680, length=336960, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell4.0.23182080.23876160 length: 694080\n",
      "[dwgsim_core] 1 sequences, total length: 694080\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 17409\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 32986 sequences (3298600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 15979)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.97, 49.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 32986 reads in 3.220 CPU sec, 3.259 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b46/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b46/filtered_read2.fastq\n",
      "[main] Real time: 3.301 sec; CPU: 3.263 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.2', allele=0, num_copies=2, relstart=112337280, refstart=31173120, length=540480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.0.112326720.113427840 length: 1101120\n",
      "[dwgsim_core] 1 sequences, total length: 1101120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 27924\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 53100 sequences (5310000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 25821)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.79, 49.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 53100 reads in 5.603 CPU sec, 5.625 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b47/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b47/filtered_read2.fastq\n",
      "[main] Real time: 5.679 sec; CPU: 5.657 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.0', allele=1, num_copies=3, relstart=49287360, refstart=31533120, length=1007040, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.1.49276800.52318080 length: 3041280\n",
      "[dwgsim_core] 1 sequences, total length: 3041280\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 78045\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 48176 sequences (4817600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48058)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.58, 49.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 12.181 CPU sec, 12.239 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 23126)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.92, 49.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 48176 reads in 5.928 CPU sec, 5.928 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read2.fastq\n",
      "[main] Real time: 18.235 sec; CPU: 18.176 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor7.0', allele=0, num_copies=2, relstart=40161600, refstart=42193920, length=627840, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor7.0.40151040.41426880 length: 1275840\n",
      "[dwgsim_core] 1 sequences, total length: 1275840\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 32438\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 61582 sequences (6158200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 30043)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.04, 49.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 61582 reads in 5.812 CPU sec, 5.836 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b49/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b49/filtered_read2.fastq\n",
      "[main] Real time: 5.906 sec; CPU: 5.878 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.3', allele=0, num_copies=2, relstart=866880, refstart=866880, length=94080, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.0.856320.1064640 length: 208320\n",
      "[dwgsim_core] 1 sequences, total length: 208320\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 6272\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 11868 sequences (1186800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 25)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 134, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 327)\n",
      "[M::mem_pestat] mean and std.dev: (139.84, 46.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_process_seqs] Processed 11868 reads in 14.259 CPU sec, 14.343 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b0/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b0/filtered_read2.fastq\n",
      "[main] Real time: 14.371 sec; CPU: 14.286 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.1', allele=1, num_copies=2, relstart=3923520, refstart=3923520, length=59520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.1.3912960.4052160 length: 139200\n",
      "[dwgsim_core] 1 sequences, total length: 139200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 3968\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 7560 sequences (756000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 7560 reads in 3.451 CPU sec, 3.488 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b1/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b1/filtered_read2.fastq\n",
      "[main] Real time: 3.509 sec; CPU: 3.473 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.1', allele=1, num_copies=2, relstart=51680640, refstart=36595200, length=2384640, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.1.51670080.56459520 length: 4789440\n",
      "[dwgsim_core] 1 sequences, total length: 4789440\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 158976\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48608)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.30, 49.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.134 CPU sec, 10.160 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48622)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.34, 49.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.213 CPU sec, 10.177 real sec\n",
      "[M::process] read 1982 sequences (198200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 48544)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.41, 49.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.244 CPU sec, 10.310 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 961)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (122, 156, 189)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (154.94, 49.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 1982 reads in 0.251 CPU sec, 0.203 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b2/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b2/filtered_read2.fastq\n",
      "[main] Real time: 30.897 sec; CPU: 30.890 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor4.5', allele=0, num_copies=2, relstart=10080000, refstart=14442240, length=486720, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor4.0.10069440.11063040 length: 993600\n",
      "[dwgsim_core] 1 sequences, total length: 993600\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 32448\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 61580 sequences (6158000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 29759)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.91, 49.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 61580 reads in 6.263 CPU sec, 6.292 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b3/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b3/filtered_read2.fastq\n",
      "[main] Real time: 6.364 sec; CPU: 6.332 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor6.0', allele=0, num_copies=2, relstart=83025600, refstart=36543360, length=80640, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor6.0.83015040.83196480 length: 181440\n",
      "[dwgsim_core] 1 sequences, total length: 181440\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5376\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 10258 sequences (1025800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 4909)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (120, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (151.13, 48.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 380)\n",
      "[M::mem_process_seqs] Processed 10258 reads in 1.122 CPU sec, 1.127 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b4/filtered_read2.fastq\n",
      "[main] Real time: 1.156 sec; CPU: 1.150 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.0', allele=0, num_copies=2, relstart=11537280, refstart=11537280, length=306240, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.0.11526720.12159360 length: 632640\n",
      "[dwgsim_core] 1 sequences, total length: 632640\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 20416\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 38790 sequences (3879000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 11025)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (150.48, 49.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 392)\n",
      "[M::mem_process_seqs] Processed 38790 reads in 4.584 CPU sec, 4.600 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b5/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b5/filtered_read2.fastq\n",
      "[main] Real time: 4.646 sec; CPU: 4.630 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.0', allele=1, num_copies=3, relstart=42415680, refstart=39857280, length=656640, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read2.fastq\n",
      "[main] Real time: 0.014 sec; CPU: 0.015 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "[sam_read1] reference 'ID:bwa\tPN:bwa\tVN:0.7.18-r1243-dirty\tCL:bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b6/filtered_read2.fastq\n",
      "' is recognized as '*'.\n",
      "[main_samview] truncated file.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.1', allele=1, num_copies=2, relstart=2054400, refstart=4494720, length=168960, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor1.1.2043840.2401920 length: 358080\n",
      "[dwgsim_core] 1 sequences, total length: 358080\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 11264\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 21430 sequences (2143000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 21430 reads in 9.946 CPU sec, 10.069 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b7/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b7/filtered_read2.fastq\n",
      "[main] Real time: 10.101 sec; CPU: 9.979 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell6.2', allele=0, num_copies=2, relstart=10119360, refstart=10119360, length=1140480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.0.10108800.12409920 length: 2301120\n",
      "[dwgsim_core] 1 sequences, total length: 2301120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 76032\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 44328 sequences (4432800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 14875)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.72, 49.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 77.299 CPU sec, 77.855 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 6622)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 149, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 319)\n",
      "[M::mem_pestat] mean and std.dev: (149.72, 49.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 387)\n",
      "[M::mem_process_seqs] Processed 44328 reads in 34.663 CPU sec, 34.852 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b8/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b8/filtered_read2.fastq\n",
      "[main] Real time: 112.774 sec; CPU: 112.029 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.1', allele=0, num_copies=3, relstart=30917760, refstart=29496960, length=8640, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.0.30907200.30953280 length: 46080\n",
      "[dwgsim_core] 1 sequences, total length: 46080\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 864\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1650 sequences (165000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 820)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 311)\n",
      "[M::mem_pestat] mean and std.dev: (148.91, 47.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 376)\n",
      "[M::mem_process_seqs] Processed 1650 reads in 0.153 CPU sec, 0.154 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b9/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b9/filtered_read2.fastq\n",
      "[main] Real time: 0.171 sec; CPU: 0.171 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.0', allele=0, num_copies=2, relstart=5328960, refstart=3700800, length=180480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.0.5318400.5699520 length: 381120\n",
      "[dwgsim_core] 1 sequences, total length: 381120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 12032\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 22808 sequences (2280800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 22808 reads in 10.277 CPU sec, 10.339 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b10/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b10/filtered_read2.fastq\n",
      "[main] Real time: 10.370 sec; CPU: 10.308 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.4', allele=0, num_copies=2, relstart=24705600, refstart=34459200, length=419520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.24695040.25554240 length: 859200\n",
      "[dwgsim_core] 1 sequences, total length: 859200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 27968\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 53146 sequences (5314600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 25750)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.11, 49.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 53146 reads in 7.219 CPU sec, 7.266 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b11/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b11/filtered_read2.fastq\n",
      "[main] Real time: 7.320 sec; CPU: 7.274 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.0', allele=1, num_copies=2, relstart=35334720, refstart=43185600, length=120000, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.1.35324160.35584320 length: 260160\n",
      "[dwgsim_core] 1 sequences, total length: 260160\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8000\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 15206 sequences (1520600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 7195)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.78, 49.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 15206 reads in 1.456 CPU sec, 1.491 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b12/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b12/filtered_read2.fastq\n",
      "[main] Real time: 1.517 sec; CPU: 1.483 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell7.3', allele=1, num_copies=2, relstart=68022720, refstart=30904320, length=26880, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell7.1.68012160.68086080 length: 73920\n",
      "[dwgsim_core] 1 sequences, total length: 73920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1792\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3422 sequences (342200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1667)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 319)\n",
      "[M::mem_pestat] mean and std.dev: (151.14, 49.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 386)\n",
      "[M::mem_process_seqs] Processed 3422 reads in 0.386 CPU sec, 0.388 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b13/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b13/filtered_read2.fastq\n",
      "[main] Real time: 0.407 sec; CPU: 0.407 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.0', allele=0, num_copies=2, relstart=14531520, refstart=14531520, length=317760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.0.14520960.15176640 length: 655680\n",
      "[dwgsim_core] 1 sequences, total length: 655680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 21184\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 40166 sequences (4016600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 19530)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.65, 49.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 40166 reads in 3.934 CPU sec, 3.956 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b14/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b14/filtered_read2.fastq\n",
      "[main] Real time: 4.000 sec; CPU: 3.979 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.1', allele=1, num_copies=2, relstart=3899520, refstart=3899520, length=290880, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.1.3888960.4490880 length: 601920\n",
      "[dwgsim_core] 1 sequences, total length: 601920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 19392\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 36802 sequences (3680200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 36802 reads in 16.989 CPU sec, 17.303 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b15/filtered_read2.fastq\n",
      "[main] Real time: 17.347 sec; CPU: 17.032 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.2', allele=0, num_copies=2, relstart=20452800, refstart=38768640, length=1080000, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.0.20442240.22622400 length: 2180160\n",
      "[dwgsim_core] 1 sequences, total length: 2180160\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 72000\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 36832 sequences (3683200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48750)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.82, 49.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 9.874 CPU sec, 9.972 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 17908)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.13, 49.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 36832 reads in 3.621 CPU sec, 3.618 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b16/filtered_read2.fastq\n",
      "[main] Real time: 13.655 sec; CPU: 13.560 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.1', allele=0, num_copies=5, relstart=11238720, refstart=11238720, length=741120, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.0.11228160.14953920 length: 3725760\n",
      "[dwgsim_core] 1 sequences, total length: 3725760\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 123520\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 32455)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.88, 49.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.194 CPU sec, 31.405 real sec\n",
      "[M::process] read 34574 sequences (3457400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 32631)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.32, 49.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.176 CPU sec, 31.370 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 11250)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.57, 49.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 34574 reads in 10.548 CPU sec, 10.574 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b17/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b17/filtered_read2.fastq\n",
      "[main] Real time: 73.425 sec; CPU: 72.988 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell3.3', allele=0, num_copies=2, relstart=29584320, refstart=21683520, length=575040, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell3.0.29573760.30744000 length: 1170240\n",
      "[dwgsim_core] 1 sequences, total length: 1170240\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 38336\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 72880 sequences (7288000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 35652)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (150.10, 49.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 72880 reads in 7.003 CPU sec, 7.064 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b18/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b18/filtered_read2.fastq\n",
      "[main] Real time: 7.138 sec; CPU: 7.077 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.0', allele=1, num_copies=2, relstart=88320, refstart=88320, length=122880, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.1.77760.343680 length: 265920\n",
      "[dwgsim_core] 1 sequences, total length: 265920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8192\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 15512 sequences (1551200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 130, 165)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (3, 273)\n",
      "[M::mem_pestat] mean and std.dev: (125.50, 33.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 327)\n",
      "[M::mem_process_seqs] Processed 15512 reads in 5.564 CPU sec, 5.606 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b19/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b19/filtered_read2.fastq\n",
      "[main] Real time: 5.644 sec; CPU: 5.597 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.2', allele=0, num_copies=2, relstart=43520640, refstart=40301760, length=2542080, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor1.0.43510080.48614400 length: 5104320\n",
      "[dwgsim_core] 1 sequences, total length: 5104320\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 169472\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48098)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.18, 49.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.475 CPU sec, 10.626 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48080)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.14, 49.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.564 CPU sec, 10.646 real sec\n",
      "[M::process] read 21522 sequences (2152200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48085)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (150.15, 49.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.474 CPU sec, 10.532 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 10338)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (150.95, 49.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 21522 reads in 2.271 CPU sec, 2.251 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b20/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b20/filtered_read2.fastq\n",
      "[main] Real time: 34.130 sec; CPU: 33.853 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.0', allele=1, num_copies=2, relstart=5947200, refstart=5947200, length=872640, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.1.5936640.7702080 length: 1765440\n",
      "[dwgsim_core] 1 sequences, total length: 1765440\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 58176\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 10344 sequences (1034400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 1, 21926)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.12, 49.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.760 CPU sec, 25.907 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2256)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 148, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (148.65, 50.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 390)\n",
      "[M::mem_process_seqs] Processed 10344 reads in 2.695 CPU sec, 2.666 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b21/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b21/filtered_read2.fastq\n",
      "[main] Real time: 28.627 sec; CPU: 28.508 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.2', allele=0, num_copies=2, relstart=22641600, refstart=11551680, length=422400, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.0.22631040.23496000 length: 864960\n",
      "[dwgsim_core] 1 sequences, total length: 864960\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 28160\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 53550 sequences (5355000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 25762)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.84, 49.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 53550 reads in 5.568 CPU sec, 5.601 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b22/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b22/filtered_read2.fastq\n",
      "[main] Real time: 5.660 sec; CPU: 5.625 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.1', allele=0, num_copies=2, relstart=9042240, refstart=9042240, length=246720, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.9031680.9545280 length: 513600\n",
      "[dwgsim_core] 1 sequences, total length: 513600\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 16448\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 31206 sequences (3120600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 15149)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.42, 49.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 31206 reads in 2.809 CPU sec, 2.846 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b23/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b23/filtered_read2.fastq\n",
      "[main] Real time: 2.887 sec; CPU: 2.849 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.2', allele=1, num_copies=4, relstart=74243520, refstart=38769600, length=393600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.1.74232960.75827520 length: 1594560\n",
      "[dwgsim_core] 1 sequences, total length: 1594560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 52480\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 99846 sequences (9984600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 1, 49001)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.46, 49.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 99846 reads in 9.303 CPU sec, 9.346 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b24/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b24/filtered_read2.fastq\n",
      "[main] Real time: 9.438 sec; CPU: 9.394 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.2', allele=1, num_copies=2, relstart=32213760, refstart=37109760, length=397440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.1.32203200.33018240 length: 815040\n",
      "[dwgsim_core] 1 sequences, total length: 815040\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 26496\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 50298 sequences (5029800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 24628)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (149.90, 49.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 50298 reads in 4.795 CPU sec, 4.817 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b25/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b25/filtered_read2.fastq\n",
      "[main] Real time: 4.878 sec; CPU: 4.852 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor8.1', allele=0, num_copies=2, relstart=40484160, refstart=37300800, length=61440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor8.0.40473600.40616640 length: 143040\n",
      "[dwgsim_core] 1 sequences, total length: 143040\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4096\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 7826 sequences (782600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 3861)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 148, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 316)\n",
      "[M::mem_pestat] mean and std.dev: (148.61, 49.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 383)\n",
      "[M::mem_process_seqs] Processed 7826 reads in 0.701 CPU sec, 0.706 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b26/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b26/filtered_read2.fastq\n",
      "[main] Real time: 0.731 sec; CPU: 0.726 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.0', allele=0, num_copies=4, relstart=25857600, refstart=32932800, length=259200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.0.25847040.26904000 length: 1056960\n",
      "[dwgsim_core] 1 sequences, total length: 1056960\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 34560\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 65512 sequences (6551200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 31882)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.35, 49.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 65512 reads in 7.567 CPU sec, 7.657 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b27/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b27/filtered_read2.fastq\n",
      "[main] Real time: 7.724 sec; CPU: 7.633 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.0', allele=0, num_copies=2, relstart=18769920, refstart=30034560, length=548160, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.0.18759360.19875840 length: 1116480\n",
      "[dwgsim_core] 1 sequences, total length: 1116480\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 36544\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 69326 sequences (6932600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 33676)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.68, 49.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 69326 reads in 7.450 CPU sec, 7.495 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b28/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b28/filtered_read2.fastq\n",
      "[main] Real time: 7.568 sec; CPU: 7.522 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor6.1', allele=1, num_copies=2, relstart=8413440, refstart=30001920, length=1089600, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor6.1.8402880.10602240 length: 2199360\n",
      "[dwgsim_core] 1 sequences, total length: 2199360\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 72640\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 38268 sequences (3826800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 48609)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.31, 49.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.997 CPU sec, 11.092 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 18635)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.24, 49.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 38268 reads in 4.269 CPU sec, 4.255 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b29/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b29/filtered_read2.fastq\n",
      "[main] Real time: 15.420 sec; CPU: 15.336 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.2', allele=0, num_copies=6, relstart=27279360, refstart=24634560, length=85440, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.0.27268800.27801600 length: 532800\n",
      "[dwgsim_core] 1 sequences, total length: 532800\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 17088\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 32548 sequences (3254800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 15636)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.85, 49.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 32548 reads in 3.463 CPU sec, 3.489 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b30/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b30/filtered_read2.fastq\n",
      "[main] Real time: 3.534 sec; CPU: 3.507 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell4.0', allele=0, num_copies=2, relstart=54717120, refstart=39722880, length=24960, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell4.0.54706560.54776640 length: 70080\n",
      "[dwgsim_core] 1 sequences, total length: 70080\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1664\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3162 sequences (316200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1559)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (113, 149, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (148.54, 50.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 3162 reads in 0.266 CPU sec, 0.267 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b31/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b31/filtered_read2.fastq\n",
      "[main] Real time: 0.288 sec; CPU: 0.287 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.0', allele=0, num_copies=2, relstart=9805440, refstart=9805440, length=60480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.0.9794880.9936000 length: 141120\n",
      "[dwgsim_core] 1 sequences, total length: 141120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 4032\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 7676 sequences (767600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2723)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 149, 180)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 306)\n",
      "[M::mem_pestat] mean and std.dev: (149.25, 48.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 369)\n",
      "[M::mem_process_seqs] Processed 7676 reads in 0.837 CPU sec, 0.843 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b32/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b32/filtered_read2.fastq\n",
      "[main] Real time: 0.866 sec; CPU: 0.860 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.1', allele=1, num_copies=2, relstart=3154560, refstart=3154560, length=265920, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.1.3144000.3696000 length: 552000\n",
      "[dwgsim_core] 1 sequences, total length: 552000\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 17728\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 33746 sequences (3374600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 33746 reads in 15.101 CPU sec, 15.256 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b33/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b33/filtered_read2.fastq\n",
      "[main] Real time: 15.297 sec; CPU: 15.142 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell1.0', allele=0, num_copies=3, relstart=44555520, refstart=14393280, length=54720, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell1.0.44544960.44729280 length: 184320\n",
      "[dwgsim_core] 1 sequences, total length: 184320\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 5472\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 10424 sequences (1042400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5045)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.29, 49.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 10424 reads in 0.994 CPU sec, 1.031 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b34/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b34/filtered_read2.fastq\n",
      "[main] Real time: 1.055 sec; CPU: 1.019 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell8.4', allele=1, num_copies=2, relstart=3094080, refstart=3094080, length=387840, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell8.1.3083520.3879360 length: 795840\n",
      "[dwgsim_core] 1 sequences, total length: 795840\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 25856\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 49136 sequences (4913600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1156)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 146, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 324)\n",
      "[M::mem_pestat] mean and std.dev: (146.83, 50.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 395)\n",
      "[M::mem_process_seqs] Processed 49136 reads in 44.335 CPU sec, 44.622 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b35/filtered_read2.fastq\n",
      "[main] Real time: 44.675 sec; CPU: 44.390 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor1.0', allele=1, num_copies=3, relstart=9312000, refstart=9312000, length=121920, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor1.1.9301440.9687360 length: 385920\n",
      "[dwgsim_core] 1 sequences, total length: 385920\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 12192\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 23108 sequences (2310800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5990)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (150.06, 49.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 392)\n",
      "[M::mem_process_seqs] Processed 23108 reads in 5.574 CPU sec, 5.610 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b36/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b36/filtered_read2.fastq\n",
      "[main] Real time: 5.642 sec; CPU: 5.608 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell4.0', allele=1, num_copies=2, relstart=38257920, refstart=15902400, length=1123200, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell4.1.38247360.40513920 length: 2266560\n",
      "[dwgsim_core] 1 sequences, total length: 2266560\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 74880\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 42576 sequences (4257600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48636)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (150.17, 49.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 11.119 CPU sec, 11.201 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 20728)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.20, 49.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 42576 reads in 4.734 CPU sec, 4.744 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b37/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b37/filtered_read2.fastq\n",
      "[main] Real time: 16.013 sec; CPU: 15.917 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell9.5', allele=1, num_copies=2, relstart=14660160, refstart=13104000, length=1711680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell9.1.14649600.18093120 length: 3443520\n",
      "[dwgsim_core] 1 sequences, total length: 3443520\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 114112\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48247)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.08, 49.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 9.061 CPU sec, 9.187 real sec\n",
      "[M::process] read 16886 sequences (1688600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48126)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.88, 49.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 9.146 CPU sec, 64.560 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 8132)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.63, 49.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 16886 reads in 1.566 CPU sec, 1.539 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b38/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b38/filtered_read2.fastq\n",
      "[main] Real time: 75.343 sec; CPU: 19.831 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor3.3', allele=0, num_copies=6, relstart=39779520, refstart=43999680, length=34560, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor3.0.39768960.39996480 length: 227520\n",
      "[dwgsim_core] 1 sequences, total length: 227520\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 6912\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 13170 sequences (1317000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 5871)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (150.59, 50.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 392)\n",
      "[M::mem_process_seqs] Processed 13170 reads in 1.235 CPU sec, 1.236 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b39/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b39/filtered_read2.fastq\n",
      "[main] Real time: 1.259 sec; CPU: 1.259 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.1', allele=0, num_copies=5, relstart=67481280, refstart=15144000, length=52800, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.0.67470720.67754880 length: 284160\n",
      "[dwgsim_core] 1 sequences, total length: 284160\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 8800\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 16738 sequences (1673800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 8310)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.93, 49.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 16738 reads in 0.964 CPU sec, 52.039 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b40/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b40/filtered_read2.fastq\n",
      "[main] Real time: 52.066 sec; CPU: 1.206 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor2.0', allele=0, num_copies=2, relstart=21788160, refstart=21788160, length=612480, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor2.0.21777600.23022720 length: 1245120\n",
      "[dwgsim_core] 1 sequences, total length: 1245120\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 40832\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 77736 sequences (7773600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 38081)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.34, 49.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 77736 reads in 6.417 CPU sec, 6.449 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b41/filtered_read2.fastq\n",
      "[main] Real time: 6.517 sec; CPU: 6.487 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell6.0', allele=1, num_copies=3, relstart=35251200, refstart=20794560, length=558720, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell6.1.35240640.36936960 length: 1696320\n",
      "[dwgsim_core] 1 sequences, total length: 1696320\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 55872\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 6154 sequences (615400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 48720)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (150.09, 49.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 8.362 CPU sec, 8.376 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 2987)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 149, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (149.16, 49.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_process_seqs] Processed 6154 reads in 0.570 CPU sec, 0.530 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b42/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b42/filtered_read2.fastq\n",
      "[main] Real time: 8.952 sec; CPU: 8.978 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell2.0', allele=1, num_copies=2, relstart=13173120, refstart=13173120, length=203520, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell2.1.13162560.13589760 length: 427200\n",
      "[dwgsim_core] 1 sequences, total length: 427200\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 13568\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 25854 sequences (2585400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 7681)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 149, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (149.60, 48.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 381)\n",
      "[M::mem_process_seqs] Processed 25854 reads in 4.611 CPU sec, 4.640 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b43/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b43/filtered_read2.fastq\n",
      "[main] Real time: 4.682 sec; CPU: 4.649 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell8.0', allele=1, num_copies=2, relstart=49497600, refstart=17547840, length=175680, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell8.1.49487040.49858560 length: 371520\n",
      "[dwgsim_core] 1 sequences, total length: 371520\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 11712\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 22314 sequences (2231400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 10978)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (151.23, 49.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 389)\n",
      "[M::mem_process_seqs] Processed 22314 reads in 1.800 CPU sec, 1.806 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b44/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b44/filtered_read2.fastq\n",
      "[main] Real time: 1.837 sec; CPU: 1.830 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell10.0', allele=1, num_copies=2, relstart=41261760, refstart=41949120, length=906240, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell10.1.41251200.43083840 length: 1832640\n",
      "[dwgsim_core] 1 sequences, total length: 1832640\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 60416\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 14686 sequences (1468600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 48018)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.16, 49.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 8.379 CPU sec, 8.387 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 7061)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (149.83, 49.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 14686 reads in 1.260 CPU sec, 1.212 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b45/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b45/filtered_read2.fastq\n",
      "[main] Real time: 9.648 sec; CPU: 9.689 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell4.1', allele=0, num_copies=2, relstart=18992640, refstart=25988160, length=160320, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell4.0.18982080.19322880 length: 340800\n",
      "[dwgsim_core] 1 sequences, total length: 340800\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 10688\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 20300 sequences (2030000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 9639)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (150.25, 49.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 388)\n",
      "[M::mem_process_seqs] Processed 20300 reads in 1.716 CPU sec, 1.717 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b46/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b46/filtered_read2.fastq\n",
      "[main] Real time: 1.745 sec; CPU: 1.745 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='cell5.2', allele=0, num_copies=2, relstart=45860160, refstart=32352960, length=29760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] cell5.0.45849600.45929280 length: 79680\n",
      "[dwgsim_core] 1 sequences, total length: 79680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 1984\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3756 sequences (375600 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 1849)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.31, 48.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 3756 reads in 0.289 CPU sec, 0.289 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b47/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b47/filtered_read2.fastq\n",
      "[main] Real time: 0.304 sec; CPU: 0.305 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.2', allele=0, num_copies=2, relstart=28458240, refstart=10897920, length=527040, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.0.28447680.29521920 length: 1074240\n",
      "[dwgsim_core] 1 sequences, total length: 1074240\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 35136\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 66834 sequences (6683400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 7320)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.51, 49.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 66834 reads in 92.649 CPU sec, 92.696 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b48/filtered_read2.fastq\n",
      "[main] Real time: 92.763 sec; CPU: 92.714 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output directory: /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/sequences\n",
      "Preparing ground truth tree...\n",
      "Initializing reference genome...\n",
      "Number of chromosomes in reference: 1\n",
      "Generating genomes, events, and profiles...\n",
      "Formating and saving profiles\n",
      "Mutation(edgeid='ancestor5.0', allele=1, num_copies=2, relstart=9013440, refstart=9013440, length=917760, d={'numcells': 10, 'founder_mult': 1, 'resolution': 960, 'block_length': 60})\n",
      "num copies: 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[dwgsim_core] ancestor5.1.9002880.10858560 length: 1855680\n",
      "[dwgsim_core] 1 sequences, total length: 1855680\n",
      "[dwgsim_core] Currently on: \n",
      "[dwgsim_core] 61184\n",
      "[dwgsim_core] Complete!\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 16418 sequences (1641800 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 0, 0, 39303)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 150, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.38, 49.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.574 CPU sec, 10.570 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 6514)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 151, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (150.55, 49.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 385)\n",
      "[M::mem_process_seqs] Processed 16418 reads in 1.774 CPU sec, 1.734 real sec\n",
      "[main] Version: 0.7.18-r1243-dirty\n",
      "[main] CMD: bwa mem /Users/rfeld/Documents/Research/SPATIAL/spatial_24/ref/chr21.fa /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b49/filtered_read1.fastq /Users/rfeld/Documents/Research/SPATIAL/output/experiments/exp22/reads/b49/filtered_read2.fastq\n",
      "[main] Real time: 12.359 sec; CPU: 12.401 sec\n",
      "[samopen] SAM header is present: 1 sequences.\n",
      "\n",
      "Usage:   samtools sort [options] <in.bam> <out.prefix>\n",
      "\n",
      "Options: -n        sort by read name\n",
      "         -f        use <out.prefix> as full file name instead of prefix\n",
      "         -o        final output to stdout\n",
      "         -l INT    compression level, from 0 to 9 [-1]\n",
      "         -@ INT    number of sorting and compression threads [1]\n",
      "         -m INT    max memory per thread; suffix K/M/G recognized [768M]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "_mutlengths = []\n",
    "_num_reads_by_cov = []\n",
    "_num_copies = []\n",
    "_signal = []\n",
    "_peaks = []\n",
    "_locs = []\n",
    "_fnames = []\n",
    "_times = []\n",
    "\n",
    "coverages = np.linspace(1, 10, 5)\n",
    "for j in range(len(coverages)):\n",
    "    mutlengths = []\n",
    "    num_reads_by_cov = []\n",
    "    num_copies = []\n",
    "    signal = []\n",
    "    peaks = []\n",
    "    locs = []\n",
    "    fnames = []\n",
    "    times = []\n",
    "    coverage = coverages[j] \n",
    "    for i in range(50):\n",
    "        start = time.time()\n",
    "        fp, aug_fp, tm, root, celldict, focal = m.genstep(seq_dict, _seqdir, log = log,\n",
    "                                                    easy = True, only_insertions=True)\n",
    "        print(tm)\n",
    "        is_insertion = (tm.num_copies != 0)\n",
    "        length = tm.length\n",
    "        rl = tm.refstart \n",
    "        rr = rl + length\n",
    "\n",
    "        mutlengths.append(length)\n",
    "        locs.append((rl, rr))\n",
    "        num_copies.append(tm.num_copies)\n",
    "        print(f\"num copies: {num_copies[-1]}\")\n",
    "        fnames.append(aug_fp)\n",
    "\n",
    "        num_reads_by_coverage_mult = coverage * tm.length * tm.num_copies // 300 \n",
    "        read_dict[\"numreads\"] = int(num_reads_by_coverage_mult)\n",
    "        num_reads_by_cov.append(num_reads_by_coverage_mult)\n",
    "\n",
    "        readdir = f\"{_readdir}/b{i}\"\n",
    "        subprocess.call([\"mkdir\", \"-p\", readdir])\n",
    "        \n",
    "        try:\n",
    "            df, sus_df = m.readstep(read_dict, readdir, f\"b{i}\", tm, aug_fp, log = log)\n",
    "            signal.append(len(df[((df[\"l\"] >= rl) & (df[\"r\"] <= rr)) & (df[\"l\"] > df[\"r\"])]))\n",
    "            peaks.append(len(df[df[\"l\"] > df[\"r\"]]))\n",
    "        except:\n",
    "            peaks.append(-1)\n",
    "            signal.append(-1)\n",
    "        \n",
    "        times.append(time.time() - start)\n",
    "    \n",
    "    _mutlengths.append(mutlengths)\n",
    "    _num_reads_by_cov.append(num_reads_by_cov)\n",
    "    _num_copies.append(_num_copies)\n",
    "    _signal.append(signal)\n",
    "    _peaks.append(peaks)\n",
    "    _locs.append(locs)\n",
    "    _fnames.append(fnames)\n",
    "    _times.append(times)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hello\n"
     ]
    }
   ],
   "source": [
    "print(\"hello\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "a_ml = np.array(_mutlengths)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "a_nr = np.asarray(_num_reads_by_cov)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[-1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1],\n",
       " [-1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1],\n",
       " [-1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1],\n",
       " [-1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1],\n",
       " [-1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1],\n",
       " [-1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1],\n",
       " [-1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1],\n",
       " [-1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1],\n",
       " [-1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1],\n",
       " [-1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1,\n",
       "  -1]]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_signal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[14], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m a_nc \u001b[38;5;241m=\u001b[39m \u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43marray\u001b[49m\u001b[43m(\u001b[49m\u001b[43m_num_copies\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[21], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m a_sig \u001b[38;5;241m=\u001b[39m \u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43marray\u001b[49m\u001b[43m(\u001b[49m\u001b[43m_num_copies\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "a_sig = np.array(_num_copies)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[9], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m a_ml \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39masarray(_mutlengths)\n\u001b[1;32m      2\u001b[0m a_nr \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39masarray(_num_reads_by_cov)\n\u001b[0;32m----> 3\u001b[0m a_nc \u001b[38;5;241m=\u001b[39m \u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43masarray\u001b[49m\u001b[43m(\u001b[49m\u001b[43m_num_copies\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      4\u001b[0m a_sig \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39masarray(_signal)\n\u001b[1;32m      5\u001b[0m a_peaks \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39masarray(_peaks)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "a_ml = np.asarray(_mutlengths)\n",
    "a_nr = np.asarray(_num_reads_by_cov)\n",
    "a_nc = np.asarray(_num_copies)\n",
    "a_sig = np.asarray(_signal)\n",
    "a_peaks = np.asarray(_peaks)\n",
    "a_times = np.asarray(_times)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a_nr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.savez('coverage_vals_exp18.npz',mutlengths=a_ml,numreads=a_nr,signals=a_sig,peaks=a_peaks,times=a_times,coverage=coverages)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "cnasim",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
